// @devvit/protos/types/devvit/dev_portal/app/info/compute_cluster.js
/**
 * #compute_cluster.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
/**
 * ComputeCluster represents the compute infrastructure cluster where an app
 * installation runs.
 */
export var ComputeCluster;
(function (ComputeCluster) {
    /** COMPUTE_CLUSTER_UNSPECIFIED - Unspecified cluster - should not be used in practice. */
    ComputeCluster[ComputeCluster["COMPUTE_CLUSTER_UNSPECIFIED"] = 0] = "COMPUTE_CLUSTER_UNSPECIFIED";
    /** COMPUTE_GO - devvit-compute-go runtime */
    ComputeCluster[ComputeCluster["COMPUTE_GO"] = 1] = "COMPUTE_GO";
    /** ELYSIUM_GCP - devvit-orchestrator, apps on GCP */
    ComputeCluster[ComputeCluster["ELYSIUM_GCP"] = 2] = "ELYSIUM_GCP";
    /** COMPUTE_GO_STAGING - devvit-compute-go staging runtime */
    ComputeCluster[ComputeCluster["COMPUTE_GO_STAGING"] = 3] = "COMPUTE_GO_STAGING";
    ComputeCluster[ComputeCluster["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ComputeCluster || (ComputeCluster = {}));
export function computeClusterFromJSON(object) {
    switch (object) {
        case 0:
        case "COMPUTE_CLUSTER_UNSPECIFIED":
            return ComputeCluster.COMPUTE_CLUSTER_UNSPECIFIED;
        case 1:
        case "COMPUTE_GO":
            return ComputeCluster.COMPUTE_GO;
        case 2:
        case "ELYSIUM_GCP":
            return ComputeCluster.ELYSIUM_GCP;
        case 3:
        case "COMPUTE_GO_STAGING":
            return ComputeCluster.COMPUTE_GO_STAGING;
        case -1:
        case "UNRECOGNIZED":
        default:
            return ComputeCluster.UNRECOGNIZED;
    }
}
export function computeClusterToJSON(object) {
    switch (object) {
        case ComputeCluster.COMPUTE_CLUSTER_UNSPECIFIED:
            return 0;
        case ComputeCluster.COMPUTE_GO:
            return 1;
        case ComputeCluster.ELYSIUM_GCP:
            return 2;
        case ComputeCluster.COMPUTE_GO_STAGING:
            return 3;
        case ComputeCluster.UNRECOGNIZED:
        default:
            return -1;
    }
}
