// @devvit/protos/json/reddit/devvit/post/v1/post.js
/**
 * #post.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
/** The height of the Post entrypoint. This is not expected to change often. */
export var EntrypointHeight;
(function (EntrypointHeight) {
    EntrypointHeight[EntrypointHeight["HEIGHT_UNSPECIFIED"] = 0] = "HEIGHT_UNSPECIFIED";
    EntrypointHeight[EntrypointHeight["REGULAR"] = 1] = "REGULAR";
    EntrypointHeight[EntrypointHeight["TALL"] = 2] = "TALL";
    EntrypointHeight[EntrypointHeight["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(EntrypointHeight || (EntrypointHeight = {}));
/**
 * Matches the values in the MediaService.MimeType enum.
 * See: https://github.snooguts.net/reddit/media-api/blob/master/api/protos/reddit/foundation/media/v1/media.proto
 */
export var ImageDataMimeType;
(function (ImageDataMimeType) {
    ImageDataMimeType[ImageDataMimeType["GIF"] = 0] = "GIF";
    ImageDataMimeType[ImageDataMimeType["JPEG"] = 1] = "JPEG";
    ImageDataMimeType[ImageDataMimeType["PNG"] = 2] = "PNG";
    ImageDataMimeType[ImageDataMimeType["WEBP"] = 3] = "WEBP";
    ImageDataMimeType[ImageDataMimeType["MP4"] = 4] = "MP4";
    ImageDataMimeType[ImageDataMimeType["QUICKTIME"] = 5] = "QUICKTIME";
    ImageDataMimeType[ImageDataMimeType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ImageDataMimeType || (ImageDataMimeType = {}));
