// @devvit/public-api/apis/reddit/models/Rule.js
var __classPrivateFieldSet = (this && this.__classPrivateFieldSet) || function (receiver, state, value, kind, f) {
    if (kind === "m") throw new TypeError("Private method is not writable");
    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a setter");
    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot write private member to an object whose class did not declare it");
    return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
};
var __classPrivateFieldGet = (this && this.__classPrivateFieldGet) || function (receiver, state, kind, f) {
    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a getter");
    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot read private member from an object whose class did not declare it");
    return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
};
var _Rule_shortName, _Rule_description, _Rule_kind, _Rule_violationReason, _Rule_priority, _Rule_createdUtc, _Rule_descriptionHtml, _Rule_subredditName, _Rule_metadata;
import { assertNonNull } from '@devvit/shared-types/NonNull.js';
import { Devvit } from '../../../devvit/Devvit.js';
import { makeGettersEnumerable } from '../helpers/makeGettersEnumerable.js';
function isKind(value) {
    return value !== undefined && (value === 'all' || value === 'link' || value === 'comment');
}
export class Rule {
    constructor(ruleData, subredditName, metadata) {
        _Rule_shortName.set(this, void 0);
        _Rule_description.set(this, void 0);
        _Rule_kind.set(this, void 0);
        _Rule_violationReason.set(this, void 0);
        _Rule_priority.set(this, void 0);
        _Rule_createdUtc.set(this, void 0);
        _Rule_descriptionHtml.set(this, void 0);
        _Rule_subredditName.set(this, void 0);
        _Rule_metadata.set(this, void 0);
        makeGettersEnumerable(this);
        const shortName = ruleData.shortName;
        const description = ruleData.description;
        const priority = ruleData.priority;
        const createdUtc = ruleData.createdUtc;
        const descriptionHtml = ruleData.descriptionHtml;
        if (!isKind(ruleData.kind)) {
            throw new Error(`Invalid kind: ${ruleData.kind}`);
        }
        assertNonNull(shortName, 'Subreddit rule is missing shortName');
        assertNonNull(description, 'Subreddit rule is missing description');
        assertNonNull(priority, 'Subreddit rule is missing priority');
        assertNonNull(createdUtc, 'Subreddit rule is missing createdUtc');
        assertNonNull(descriptionHtml, 'Subreddit rule is missing descriptionHtml');
        __classPrivateFieldSet(this, _Rule_shortName, shortName, "f");
        __classPrivateFieldSet(this, _Rule_description, description, "f");
        __classPrivateFieldSet(this, _Rule_kind, ruleData.kind, "f");
        __classPrivateFieldSet(this, _Rule_violationReason, ruleData.violationReason ?? shortName, "f");
        __classPrivateFieldSet(this, _Rule_priority, priority, "f");
        __classPrivateFieldSet(this, _Rule_createdUtc, createdUtc, "f");
        __classPrivateFieldSet(this, _Rule_descriptionHtml, descriptionHtml, "f");
        __classPrivateFieldSet(this, _Rule_subredditName, subredditName.startsWith('r/') ? subredditName.slice(2) : subredditName, "f");
        __classPrivateFieldSet(this, _Rule_metadata, metadata, "f");
    }
    /**
     * The name for the rule.
     */
    get shortName() {
        return __classPrivateFieldGet(this, _Rule_shortName, "f");
    }
    /**
     * The full description of the rule. This appears on your subreddit's sidebar.
     */
    get description() {
        return __classPrivateFieldGet(this, _Rule_description, "f");
    }
    /**
     * Which Reddit objects this rule applies to. One of "all", "link" (AKA posts), "comment".
     */
    get kind() {
        return __classPrivateFieldGet(this, _Rule_kind, "f");
    }
    /**
     * Text to show users when reporting content due to this rule. Defaults to the shortName.
     */
    get violationReason() {
        return __classPrivateFieldGet(this, _Rule_violationReason, "f");
    }
    /**
     * The zero-indexed rank of the rule on the subreddit sidebar. Lower numbers appear on top.
     */
    get priority() {
        return __classPrivateFieldGet(this, _Rule_priority, "f");
    }
    /**
     * The Unix timestamp of when the rule was created.
     */
    get createdUtc() {
        return __classPrivateFieldGet(this, _Rule_createdUtc, "f");
    }
    get descriptionHtml() {
        return __classPrivateFieldGet(this, _Rule_descriptionHtml, "f");
    }
    /**
     * The name (without r/ prefix) of the subreddit the rule belongs to.
     */
    get subredditName() {
        return __classPrivateFieldGet(this, _Rule_subredditName, "f");
    }
    toJSON() {
        return {
            shortName: this.shortName,
            description: this.description,
            kind: this.kind,
            violationReason: this.violationReason,
            priority: this.priority,
            createdUtc: this.createdUtc,
            subredditName: this.subredditName,
            ...(this.descriptionHtml !== undefined && { descriptionHtml: this.descriptionHtml }),
        };
    }
    /**
     * Update an existing rule.
     *
     * @param options - New values for an existing rule. All fields are optional. If a field is not provided, the existing value will not be changed.
     * @param options.shortName - New name for the rule. The rule name must be unique within this subreddit.
     * @param options.description - New full description of the rule.
     * @param options.kind - Which Reddit objects this rule should apply to now. One of "all", "link", "comment".
     * @param options.violationReason - New text to show users when reporting content due to this rule.
     *   To reset violationReason to the default value of shortName, pass the shortName or an empty string.
     */
    async update(options) {
        const client = Devvit.redditAPIPlugins.Subreddits;
        await client.UpdateSubredditRule({
            r: this.subredditName,
            oldShortName: this.shortName,
            shortName: options.shortName ?? this.shortName, // The shortName is required even if it's not changing
            description: options.description ?? '', // Leaving the description empty will not update it
            kind: options.kind ?? this.kind, // The kind is required even if it's not changing
            // The Reddit API doesn't provide a way to reset the violation reason, so we treat an explicit empty string as a signal to reset it to the shortName
            violationReason: options.violationReason === '' ? this.shortName : (options.violationReason ?? ''),
        }, __classPrivateFieldGet(this, _Rule_metadata, "f"));
    }
    async delete() {
        return Rule.delete(this.subredditName, this.shortName, __classPrivateFieldGet(this, _Rule_metadata, "f"));
    }
    static async getRules(subredditName, metadata) {
        const client = Devvit.redditAPIPlugins.Subreddits;
        const response = await client.SubredditAboutRules({ subreddit: subredditName }, metadata);
        return (response.rules ?? []).map((ruleData) => new Rule(ruleData, subredditName, metadata));
    }
    static async createRule(subredditName, options, metadata) {
        if (options.shortName === '') {
            throw new Error('Short name cannot be empty');
        }
        if (options.description === '') {
            throw new Error('Description cannot be empty');
        }
        const client = Devvit.redditAPIPlugins.Subreddits;
        await client.AddSubredditRule({
            r: subredditName,
            description: options.description,
            kind: options.kind,
            shortName: options.shortName,
            violationReason: options.violationReason ?? '',
        }, metadata);
    }
    static async reorderRules(subredditName, rules, metadata) {
        const client = Devvit.redditAPIPlugins.Subreddits;
        await client.ReorderSubredditRules({
            r: subredditName,
            newRuleOrder: rules.map((rule) => rule.shortName).join(','),
        }, metadata);
    }
    static async delete(subredditName, shortName, metadata) {
        const client = Devvit.redditAPIPlugins.Subreddits;
        await client.RemoveSubredditRule({
            r: subredditName,
            shortName,
        }, metadata);
    }
}
_Rule_shortName = new WeakMap(), _Rule_description = new WeakMap(), _Rule_kind = new WeakMap(), _Rule_violationReason = new WeakMap(), _Rule_priority = new WeakMap(), _Rule_createdUtc = new WeakMap(), _Rule_descriptionHtml = new WeakMap(), _Rule_subredditName = new WeakMap(), _Rule_metadata = new WeakMap();
