// @devvit/protos/types/devvit/plugin_svc/userauth/userauth_svc.twirp.js
import { TwirpServer, TwirpError, TwirpErrorCode, TwirpContentType, chainInterceptors, } from 'twirp-ts';
import { EnsureRefreshTokenExistsRequest, EnsureRefreshTokenExistsResponse, } from './userauth_msg.js';
export var UserAuthMethod;
(function (UserAuthMethod) {
    UserAuthMethod["EnsureRefreshTokenExists"] = "EnsureRefreshTokenExists";
})(UserAuthMethod || (UserAuthMethod = {}));
export const UserAuthMethodList = [UserAuthMethod.EnsureRefreshTokenExists];
export function createUserAuthServer(service) {
    return new TwirpServer({
        service,
        packageName: 'devvit.plugin_svc.userauth',
        serviceName: 'UserAuth',
        methodList: UserAuthMethodList,
        matchRoute: matchUserAuthRoute,
    });
}
function matchUserAuthRoute(method, events) {
    switch (method) {
        case 'EnsureRefreshTokenExists':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'EnsureRefreshTokenExists' };
                await events.onMatch(ctx);
                return handleUserAuthEnsureRefreshTokenExistsRequest(ctx, service, data, interceptors);
            };
        default:
            events.onNotFound();
            const msg = `no handler found`;
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleUserAuthEnsureRefreshTokenExistsRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleUserAuthEnsureRefreshTokenExistsJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleUserAuthEnsureRefreshTokenExistsProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
async function handleUserAuthEnsureRefreshTokenExistsJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = EnsureRefreshTokenExistsRequest.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.EnsureRefreshTokenExists(ctx, inputReq);
        });
    }
    else {
        response = await service.EnsureRefreshTokenExists(ctx, request);
    }
    return JSON.stringify(EnsureRefreshTokenExistsResponse.toJSON(response));
}
async function handleUserAuthEnsureRefreshTokenExistsProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = EnsureRefreshTokenExistsRequest.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.EnsureRefreshTokenExists(ctx, inputReq);
        });
    }
    else {
        response = await service.EnsureRefreshTokenExists(ctx, request);
    }
    return Buffer.from(EnsureRefreshTokenExistsResponse.encode(response).finish());
}
