// @devvit/protos/types/devvit/plugin_svc/userauth/userauth_msg.js
/**
 * #userauth_msg.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import _m0 from "protobufjs/minimal.js";
import { messageTypeRegistry } from "../../../typeRegistry.js";
function createBaseEnsureRefreshTokenExistsRequest() {
    return { userId: "" };
}
export const EnsureRefreshTokenExistsRequest = {
    $type: "devvit.plugin_svc.userauth.EnsureRefreshTokenExistsRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.userId !== "") {
            writer.uint32(10).string(message.userId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEnsureRefreshTokenExistsRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.userId = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { userId: isSet(object.userId) ? globalThis.String(object.userId) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.userId !== "") {
            obj.userId = message.userId;
        }
        return obj;
    },
    create(base) {
        return EnsureRefreshTokenExistsRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseEnsureRefreshTokenExistsRequest();
        message.userId = object.userId ?? "";
        return message;
    },
};
messageTypeRegistry.set(EnsureRefreshTokenExistsRequest.$type, EnsureRefreshTokenExistsRequest);
function createBaseEnsureRefreshTokenExistsResponse() {
    return {};
}
export const EnsureRefreshTokenExistsResponse = {
    $type: "devvit.plugin_svc.userauth.EnsureRefreshTokenExistsResponse",
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseEnsureRefreshTokenExistsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return EnsureRefreshTokenExistsResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseEnsureRefreshTokenExistsResponse();
        return message;
    },
};
messageTypeRegistry.set(EnsureRefreshTokenExistsResponse.$type, EnsureRefreshTokenExistsResponse);
function isSet(value) {
    return value !== null && value !== undefined;
}
