// @devvit/protos/types/devvit/plugin_svc/userauth/userauth_svc.js
/**
 * #userauth_svc.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import _m0 from "protobufjs/minimal.js";
import { EnsureRefreshTokenExistsRequest, EnsureRefreshTokenExistsResponse } from "./userauth_msg.js";
export const UserAuthServiceName = "devvit.plugin_svc.userauth.UserAuth";
export class UserAuthClientImpl {
    constructor(rpc, opts) {
        this.service = opts?.service || UserAuthServiceName;
        this.rpc = rpc;
        this.EnsureRefreshTokenExists = this.EnsureRefreshTokenExists.bind(this);
    }
    EnsureRefreshTokenExists(request, metadata) {
        const data = EnsureRefreshTokenExistsRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "EnsureRefreshTokenExists", data, metadata);
        return promise.then((data) => EnsureRefreshTokenExistsResponse.decode(_m0.Reader.create(data)));
    }
}
export const UserAuthDefinition = {
    name: "UserAuth",
    fullName: "devvit.plugin_svc.userauth.UserAuth",
    methods: {
        /**
         * EnsureRefreshTokenExists checks the shared refresh-token cache for the
         * given user and, if absent, may mint a fresh refresh token. Intended to be invoked from a
         * context that carries the user's verified edge context.
         */
        ensureRefreshTokenExists: {
            name: "EnsureRefreshTokenExists",
            requestType: EnsureRefreshTokenExistsRequest,
            requestStream: false,
            responseType: EnsureRefreshTokenExistsResponse,
            responseStream: false,
            options: {},
        },
    },
};
