// @devvit/protos/types/devvit/plugin/notifications/gamesdrawerbadge_msg.js
/**
 * #gamesdrawerbadge_msg.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import _m0 from "protobufjs/minimal.js";
import { Timestamp } from "../../../google/protobuf/timestamp.js";
import { messageTypeRegistry } from "../../../typeRegistry.js";
function createBaseShowGameBadgeRequest() {
    return { post: "", expiresAt: undefined };
}
export const ShowGameBadgeRequest = {
    $type: "devvit.plugin.notifications.ShowGameBadgeRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.post !== "") {
            writer.uint32(10).string(message.post);
        }
        if (message.expiresAt !== undefined) {
            Timestamp.encode(toTimestamp(message.expiresAt), writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseShowGameBadgeRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.post = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.expiresAt = fromTimestamp(Timestamp.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            post: isSet(object.post) ? globalThis.String(object.post) : "",
            expiresAt: isSet(object.expiresAt) ? fromJsonTimestamp(object.expiresAt) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.post !== "") {
            obj.post = message.post;
        }
        if (message.expiresAt !== undefined) {
            obj.expiresAt = message.expiresAt.toISOString();
        }
        return obj;
    },
    create(base) {
        return ShowGameBadgeRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseShowGameBadgeRequest();
        message.post = object.post ?? "";
        message.expiresAt = object.expiresAt ?? undefined;
        return message;
    },
};
messageTypeRegistry.set(ShowGameBadgeRequest.$type, ShowGameBadgeRequest);
function createBaseShowGameBadgeResponse() {
    return { success: false, message: undefined };
}
export const ShowGameBadgeResponse = {
    $type: "devvit.plugin.notifications.ShowGameBadgeResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.success !== false) {
            writer.uint32(8).bool(message.success);
        }
        if (message.message !== undefined) {
            writer.uint32(18).string(message.message);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseShowGameBadgeResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.success = reader.bool();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.message = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            success: isSet(object.success) ? globalThis.Boolean(object.success) : false,
            message: isSet(object.message) ? globalThis.String(object.message) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.success !== false) {
            obj.success = message.success;
        }
        if (message.message !== undefined) {
            obj.message = message.message;
        }
        return obj;
    },
    create(base) {
        return ShowGameBadgeResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseShowGameBadgeResponse();
        message.success = object.success ?? false;
        message.message = object.message ?? undefined;
        return message;
    },
};
messageTypeRegistry.set(ShowGameBadgeResponse.$type, ShowGameBadgeResponse);
function createBaseDismissGameBadgeRequest() {
    return {};
}
export const DismissGameBadgeRequest = {
    $type: "devvit.plugin.notifications.DismissGameBadgeRequest",
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDismissGameBadgeRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return DismissGameBadgeRequest.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseDismissGameBadgeRequest();
        return message;
    },
};
messageTypeRegistry.set(DismissGameBadgeRequest.$type, DismissGameBadgeRequest);
function createBaseDismissGameBadgeResponse() {
    return { success: false };
}
export const DismissGameBadgeResponse = {
    $type: "devvit.plugin.notifications.DismissGameBadgeResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.success !== false) {
            writer.uint32(8).bool(message.success);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDismissGameBadgeResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.success = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { success: isSet(object.success) ? globalThis.Boolean(object.success) : false };
    },
    toJSON(message) {
        const obj = {};
        if (message.success !== false) {
            obj.success = message.success;
        }
        return obj;
    },
    create(base) {
        return DismissGameBadgeResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseDismissGameBadgeResponse();
        message.success = object.success ?? false;
        return message;
    },
};
messageTypeRegistry.set(DismissGameBadgeResponse.$type, DismissGameBadgeResponse);
function createBaseGetGameBadgeStatusRequest() {
    return {};
}
export const GetGameBadgeStatusRequest = {
    $type: "devvit.plugin.notifications.GetGameBadgeStatusRequest",
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetGameBadgeStatusRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return GetGameBadgeStatusRequest.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseGetGameBadgeStatusRequest();
        return message;
    },
};
messageTypeRegistry.set(GetGameBadgeStatusRequest.$type, GetGameBadgeStatusRequest);
function createBaseGetGameBadgeStatusResponse() {
    return { hasActiveBadge: false, expiresAt: undefined, post: undefined };
}
export const GetGameBadgeStatusResponse = {
    $type: "devvit.plugin.notifications.GetGameBadgeStatusResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.hasActiveBadge !== false) {
            writer.uint32(8).bool(message.hasActiveBadge);
        }
        if (message.expiresAt !== undefined) {
            Timestamp.encode(toTimestamp(message.expiresAt), writer.uint32(18).fork()).ldelim();
        }
        if (message.post !== undefined) {
            writer.uint32(26).string(message.post);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetGameBadgeStatusResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.hasActiveBadge = reader.bool();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.expiresAt = fromTimestamp(Timestamp.decode(reader, reader.uint32()));
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.post = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            hasActiveBadge: isSet(object.hasActiveBadge) ? globalThis.Boolean(object.hasActiveBadge) : false,
            expiresAt: isSet(object.expiresAt) ? fromJsonTimestamp(object.expiresAt) : undefined,
            post: isSet(object.post) ? globalThis.String(object.post) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.hasActiveBadge !== false) {
            obj.hasActiveBadge = message.hasActiveBadge;
        }
        if (message.expiresAt !== undefined) {
            obj.expiresAt = message.expiresAt.toISOString();
        }
        if (message.post !== undefined) {
            obj.post = message.post;
        }
        return obj;
    },
    create(base) {
        return GetGameBadgeStatusResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetGameBadgeStatusResponse();
        message.hasActiveBadge = object.hasActiveBadge ?? false;
        message.expiresAt = object.expiresAt ?? undefined;
        message.post = object.post ?? undefined;
        return message;
    },
};
messageTypeRegistry.set(GetGameBadgeStatusResponse.$type, GetGameBadgeStatusResponse);
function createBaseGetGameBadgeStatusForInstallationsRequest() {
    return { installationIds: [] };
}
export const GetGameBadgeStatusForInstallationsRequest = {
    $type: "devvit.plugin.notifications.GetGameBadgeStatusForInstallationsRequest",
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.installationIds) {
            writer.uint32(10).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetGameBadgeStatusForInstallationsRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationIds.push(reader.string());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationIds: globalThis.Array.isArray(object?.installationIds)
                ? object.installationIds.map((e) => globalThis.String(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationIds?.length) {
            obj.installationIds = message.installationIds;
        }
        return obj;
    },
    create(base) {
        return GetGameBadgeStatusForInstallationsRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetGameBadgeStatusForInstallationsRequest();
        message.installationIds = object.installationIds?.map((e) => e) || [];
        return message;
    },
};
messageTypeRegistry.set(GetGameBadgeStatusForInstallationsRequest.$type, GetGameBadgeStatusForInstallationsRequest);
function createBaseGetGameBadgeStatusForInstallationsResponse() {
    return { statuses: {} };
}
export const GetGameBadgeStatusForInstallationsResponse = {
    $type: "devvit.plugin.notifications.GetGameBadgeStatusForInstallationsResponse",
    encode(message, writer = _m0.Writer.create()) {
        Object.entries(message.statuses).forEach(([key, value]) => {
            GetGameBadgeStatusForInstallationsResponse_StatusesEntry.encode({ key: key, value }, writer.uint32(10).fork()).ldelim();
        });
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetGameBadgeStatusForInstallationsResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    const entry1 = GetGameBadgeStatusForInstallationsResponse_StatusesEntry.decode(reader, reader.uint32());
                    if (entry1.value !== undefined) {
                        message.statuses[entry1.key] = entry1.value;
                    }
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            statuses: isObject(object.statuses)
                ? Object.entries(object.statuses).reduce((acc, [key, value]) => {
                    acc[key] = GetGameBadgeStatusResponse.fromJSON(value);
                    return acc;
                }, {})
                : {},
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.statuses) {
            const entries = Object.entries(message.statuses);
            if (entries.length > 0) {
                obj.statuses = {};
                entries.forEach(([k, v]) => {
                    obj.statuses[k] = GetGameBadgeStatusResponse.toJSON(v);
                });
            }
        }
        return obj;
    },
    create(base) {
        return GetGameBadgeStatusForInstallationsResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetGameBadgeStatusForInstallationsResponse();
        message.statuses = Object.entries(object.statuses ?? {}).reduce((acc, [key, value]) => {
            if (value !== undefined) {
                acc[key] = GetGameBadgeStatusResponse.fromPartial(value);
            }
            return acc;
        }, {});
        return message;
    },
};
messageTypeRegistry.set(GetGameBadgeStatusForInstallationsResponse.$type, GetGameBadgeStatusForInstallationsResponse);
function createBaseGetGameBadgeStatusForInstallationsResponse_StatusesEntry() {
    return { key: "", value: undefined };
}
export const GetGameBadgeStatusForInstallationsResponse_StatusesEntry = {
    $type: "devvit.plugin.notifications.GetGameBadgeStatusForInstallationsResponse.StatusesEntry",
    encode(message, writer = _m0.Writer.create()) {
        if (message.key !== "") {
            writer.uint32(10).string(message.key);
        }
        if (message.value !== undefined) {
            GetGameBadgeStatusResponse.encode(message.value, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetGameBadgeStatusForInstallationsResponse_StatusesEntry();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.key = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.value = GetGameBadgeStatusResponse.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            key: isSet(object.key) ? globalThis.String(object.key) : "",
            value: isSet(object.value) ? GetGameBadgeStatusResponse.fromJSON(object.value) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.key !== "") {
            obj.key = message.key;
        }
        if (message.value !== undefined) {
            obj.value = GetGameBadgeStatusResponse.toJSON(message.value);
        }
        return obj;
    },
    create(base) {
        return GetGameBadgeStatusForInstallationsResponse_StatusesEntry.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetGameBadgeStatusForInstallationsResponse_StatusesEntry();
        message.key = object.key ?? "";
        message.value = (object.value !== undefined && object.value !== null)
            ? GetGameBadgeStatusResponse.fromPartial(object.value)
            : undefined;
        return message;
    },
};
messageTypeRegistry.set(GetGameBadgeStatusForInstallationsResponse_StatusesEntry.$type, GetGameBadgeStatusForInstallationsResponse_StatusesEntry);
function toTimestamp(date) {
    const seconds = Math.trunc(date.getTime() / 1000);
    const nanos = (date.getTime() % 1000) * 1000000;
    return { seconds, nanos };
}
function fromTimestamp(t) {
    let millis = (t.seconds || 0) * 1000;
    millis += (t.nanos || 0) / 1000000;
    return new globalThis.Date(millis);
}
function fromJsonTimestamp(o) {
    if (o instanceof globalThis.Date) {
        return o;
    }
    else if (typeof o === "string") {
        return new globalThis.Date(o);
    }
    else {
        return fromTimestamp(Timestamp.fromJSON(o));
    }
}
function isObject(value) {
    return typeof value === "object" && value !== null;
}
function isSet(value) {
    return value !== null && value !== undefined;
}
