// @devvit/protos/types/devvit/plugin/notifications/notifications_svc.twirp.js
import { TwirpServer, TwirpError, TwirpErrorCode, TwirpContentType, chainInterceptors, } from 'twirp-ts';
import { Empty } from '../../../google/protobuf/empty.js';
import { OptInCurrentUserResponse, OptOutCurrentUserResponse, ListOptedInUsersRequest, ListOptedInUsersResponse, IsOptedInRequest, IsOptedInResponse, } from './optin_msg.js';
import { EnqueueRequest, EnqueueResponse } from './pushnotif_msg.js';
import { ShowGameBadgeRequest, ShowGameBadgeResponse, DismissGameBadgeRequest, DismissGameBadgeResponse, GetGameBadgeStatusRequest, GetGameBadgeStatusResponse, } from './gamesdrawerbadge_msg.js';
export var NotificationsMethod;
(function (NotificationsMethod) {
    NotificationsMethod["OptInCurrentUser"] = "OptInCurrentUser";
    NotificationsMethod["OptOutCurrentUser"] = "OptOutCurrentUser";
    NotificationsMethod["ListOptedInUsers"] = "ListOptedInUsers";
    NotificationsMethod["IsOptedIn"] = "IsOptedIn";
    NotificationsMethod["Enqueue"] = "Enqueue";
    NotificationsMethod["ShowGameBadge"] = "ShowGameBadge";
    NotificationsMethod["DismissGameBadge"] = "DismissGameBadge";
    NotificationsMethod["GetGameBadgeStatus"] = "GetGameBadgeStatus";
})(NotificationsMethod || (NotificationsMethod = {}));
export const NotificationsMethodList = [
    NotificationsMethod.OptInCurrentUser,
    NotificationsMethod.OptOutCurrentUser,
    NotificationsMethod.ListOptedInUsers,
    NotificationsMethod.IsOptedIn,
    NotificationsMethod.Enqueue,
    NotificationsMethod.ShowGameBadge,
    NotificationsMethod.DismissGameBadge,
    NotificationsMethod.GetGameBadgeStatus,
];
export function createNotificationsServer(service) {
    return new TwirpServer({
        service,
        packageName: 'devvit.plugin.notifications',
        serviceName: 'Notifications',
        methodList: NotificationsMethodList,
        matchRoute: matchNotificationsRoute,
    });
}
function matchNotificationsRoute(method, events) {
    switch (method) {
        case 'OptInCurrentUser':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'OptInCurrentUser' };
                await events.onMatch(ctx);
                return handleNotificationsOptInCurrentUserRequest(ctx, service, data, interceptors);
            };
        case 'OptOutCurrentUser':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'OptOutCurrentUser' };
                await events.onMatch(ctx);
                return handleNotificationsOptOutCurrentUserRequest(ctx, service, data, interceptors);
            };
        case 'ListOptedInUsers':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'ListOptedInUsers' };
                await events.onMatch(ctx);
                return handleNotificationsListOptedInUsersRequest(ctx, service, data, interceptors);
            };
        case 'IsOptedIn':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'IsOptedIn' };
                await events.onMatch(ctx);
                return handleNotificationsIsOptedInRequest(ctx, service, data, interceptors);
            };
        case 'Enqueue':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'Enqueue' };
                await events.onMatch(ctx);
                return handleNotificationsEnqueueRequest(ctx, service, data, interceptors);
            };
        case 'ShowGameBadge':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'ShowGameBadge' };
                await events.onMatch(ctx);
                return handleNotificationsShowGameBadgeRequest(ctx, service, data, interceptors);
            };
        case 'DismissGameBadge':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'DismissGameBadge' };
                await events.onMatch(ctx);
                return handleNotificationsDismissGameBadgeRequest(ctx, service, data, interceptors);
            };
        case 'GetGameBadgeStatus':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'GetGameBadgeStatus' };
                await events.onMatch(ctx);
                return handleNotificationsGetGameBadgeStatusRequest(ctx, service, data, interceptors);
            };
        default:
            events.onNotFound();
            const msg = `no handler found`;
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleNotificationsOptInCurrentUserRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleNotificationsOptInCurrentUserJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleNotificationsOptInCurrentUserProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleNotificationsOptOutCurrentUserRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleNotificationsOptOutCurrentUserJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleNotificationsOptOutCurrentUserProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleNotificationsListOptedInUsersRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleNotificationsListOptedInUsersJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleNotificationsListOptedInUsersProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleNotificationsIsOptedInRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleNotificationsIsOptedInJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleNotificationsIsOptedInProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleNotificationsEnqueueRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleNotificationsEnqueueJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleNotificationsEnqueueProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleNotificationsShowGameBadgeRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleNotificationsShowGameBadgeJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleNotificationsShowGameBadgeProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleNotificationsDismissGameBadgeRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleNotificationsDismissGameBadgeJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleNotificationsDismissGameBadgeProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleNotificationsGetGameBadgeStatusRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleNotificationsGetGameBadgeStatusJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleNotificationsGetGameBadgeStatusProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
async function handleNotificationsOptInCurrentUserJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = Empty.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.OptInCurrentUser(ctx, inputReq);
        });
    }
    else {
        response = await service.OptInCurrentUser(ctx, request);
    }
    return JSON.stringify(OptInCurrentUserResponse.toJSON(response));
}
async function handleNotificationsOptOutCurrentUserJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = Empty.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.OptOutCurrentUser(ctx, inputReq);
        });
    }
    else {
        response = await service.OptOutCurrentUser(ctx, request);
    }
    return JSON.stringify(OptOutCurrentUserResponse.toJSON(response));
}
async function handleNotificationsListOptedInUsersJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = ListOptedInUsersRequest.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.ListOptedInUsers(ctx, inputReq);
        });
    }
    else {
        response = await service.ListOptedInUsers(ctx, request);
    }
    return JSON.stringify(ListOptedInUsersResponse.toJSON(response));
}
async function handleNotificationsIsOptedInJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = IsOptedInRequest.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.IsOptedIn(ctx, inputReq);
        });
    }
    else {
        response = await service.IsOptedIn(ctx, request);
    }
    return JSON.stringify(IsOptedInResponse.toJSON(response));
}
async function handleNotificationsEnqueueJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = EnqueueRequest.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.Enqueue(ctx, inputReq);
        });
    }
    else {
        response = await service.Enqueue(ctx, request);
    }
    return JSON.stringify(EnqueueResponse.toJSON(response));
}
async function handleNotificationsShowGameBadgeJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = ShowGameBadgeRequest.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.ShowGameBadge(ctx, inputReq);
        });
    }
    else {
        response = await service.ShowGameBadge(ctx, request);
    }
    return JSON.stringify(ShowGameBadgeResponse.toJSON(response));
}
async function handleNotificationsDismissGameBadgeJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = DismissGameBadgeRequest.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.DismissGameBadge(ctx, inputReq);
        });
    }
    else {
        response = await service.DismissGameBadge(ctx, request);
    }
    return JSON.stringify(DismissGameBadgeResponse.toJSON(response));
}
async function handleNotificationsGetGameBadgeStatusJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = GetGameBadgeStatusRequest.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.GetGameBadgeStatus(ctx, inputReq);
        });
    }
    else {
        response = await service.GetGameBadgeStatus(ctx, request);
    }
    return JSON.stringify(GetGameBadgeStatusResponse.toJSON(response));
}
async function handleNotificationsOptInCurrentUserProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = Empty.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.OptInCurrentUser(ctx, inputReq);
        });
    }
    else {
        response = await service.OptInCurrentUser(ctx, request);
    }
    return Buffer.from(OptInCurrentUserResponse.encode(response).finish());
}
async function handleNotificationsOptOutCurrentUserProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = Empty.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.OptOutCurrentUser(ctx, inputReq);
        });
    }
    else {
        response = await service.OptOutCurrentUser(ctx, request);
    }
    return Buffer.from(OptOutCurrentUserResponse.encode(response).finish());
}
async function handleNotificationsListOptedInUsersProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = ListOptedInUsersRequest.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.ListOptedInUsers(ctx, inputReq);
        });
    }
    else {
        response = await service.ListOptedInUsers(ctx, request);
    }
    return Buffer.from(ListOptedInUsersResponse.encode(response).finish());
}
async function handleNotificationsIsOptedInProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = IsOptedInRequest.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.IsOptedIn(ctx, inputReq);
        });
    }
    else {
        response = await service.IsOptedIn(ctx, request);
    }
    return Buffer.from(IsOptedInResponse.encode(response).finish());
}
async function handleNotificationsEnqueueProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = EnqueueRequest.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.Enqueue(ctx, inputReq);
        });
    }
    else {
        response = await service.Enqueue(ctx, request);
    }
    return Buffer.from(EnqueueResponse.encode(response).finish());
}
async function handleNotificationsShowGameBadgeProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = ShowGameBadgeRequest.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.ShowGameBadge(ctx, inputReq);
        });
    }
    else {
        response = await service.ShowGameBadge(ctx, request);
    }
    return Buffer.from(ShowGameBadgeResponse.encode(response).finish());
}
async function handleNotificationsDismissGameBadgeProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = DismissGameBadgeRequest.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.DismissGameBadge(ctx, inputReq);
        });
    }
    else {
        response = await service.DismissGameBadge(ctx, request);
    }
    return Buffer.from(DismissGameBadgeResponse.encode(response).finish());
}
async function handleNotificationsGetGameBadgeStatusProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = GetGameBadgeStatusRequest.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.GetGameBadgeStatus(ctx, inputReq);
        });
    }
    else {
        response = await service.GetGameBadgeStatus(ctx, request);
    }
    return Buffer.from(GetGameBadgeStatusResponse.encode(response).finish());
}
