// @devvit/protos/types/devvit/plugin/notifications/notifications_svc.twirp-client.js
import { Empty } from '../../../google/protobuf/empty.js';
import { OptInCurrentUserResponse, OptOutCurrentUserResponse, ListOptedInUsersRequest, ListOptedInUsersResponse, IsOptedInRequest, IsOptedInResponse, } from './optin_msg.js';
import { EnqueueRequest, EnqueueResponse } from './pushnotif_msg.js';
import { ShowGameBadgeRequest, ShowGameBadgeResponse, DismissGameBadgeRequest, DismissGameBadgeResponse, GetGameBadgeStatusRequest, GetGameBadgeStatusResponse, } from './gamesdrawerbadge_msg.js';
export class NotificationsClientJSON {
    constructor(rpc) {
        this.rpc = rpc;
        this.OptInCurrentUser.bind(this);
        this.OptOutCurrentUser.bind(this);
        this.ListOptedInUsers.bind(this);
        this.IsOptedIn.bind(this);
        this.Enqueue.bind(this);
        this.ShowGameBadge.bind(this);
        this.DismissGameBadge.bind(this);
        this.GetGameBadgeStatus.bind(this);
    }
    OptInCurrentUser(request) {
        const data = Empty.toJSON(request);
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'OptInCurrentUser', 'application/json', data);
        return promise.then((data) => OptInCurrentUserResponse.fromJSON(data));
    }
    OptOutCurrentUser(request) {
        const data = Empty.toJSON(request);
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'OptOutCurrentUser', 'application/json', data);
        return promise.then((data) => OptOutCurrentUserResponse.fromJSON(data));
    }
    ListOptedInUsers(request) {
        const data = ListOptedInUsersRequest.toJSON(request);
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'ListOptedInUsers', 'application/json', data);
        return promise.then((data) => ListOptedInUsersResponse.fromJSON(data));
    }
    IsOptedIn(request) {
        const data = IsOptedInRequest.toJSON(request);
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'IsOptedIn', 'application/json', data);
        return promise.then((data) => IsOptedInResponse.fromJSON(data));
    }
    Enqueue(request) {
        const data = EnqueueRequest.toJSON(request);
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'Enqueue', 'application/json', data);
        return promise.then((data) => EnqueueResponse.fromJSON(data));
    }
    ShowGameBadge(request) {
        const data = ShowGameBadgeRequest.toJSON(request);
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'ShowGameBadge', 'application/json', data);
        return promise.then((data) => ShowGameBadgeResponse.fromJSON(data));
    }
    DismissGameBadge(request) {
        const data = DismissGameBadgeRequest.toJSON(request);
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'DismissGameBadge', 'application/json', data);
        return promise.then((data) => DismissGameBadgeResponse.fromJSON(data));
    }
    GetGameBadgeStatus(request) {
        const data = GetGameBadgeStatusRequest.toJSON(request);
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'GetGameBadgeStatus', 'application/json', data);
        return promise.then((data) => GetGameBadgeStatusResponse.fromJSON(data));
    }
}
export class NotificationsClientProtobuf {
    constructor(rpc) {
        this.rpc = rpc;
        this.OptInCurrentUser.bind(this);
        this.OptOutCurrentUser.bind(this);
        this.ListOptedInUsers.bind(this);
        this.IsOptedIn.bind(this);
        this.Enqueue.bind(this);
        this.ShowGameBadge.bind(this);
        this.DismissGameBadge.bind(this);
        this.GetGameBadgeStatus.bind(this);
    }
    OptInCurrentUser(request) {
        const data = Empty.encode(request).finish();
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'OptInCurrentUser', 'application/protobuf', data);
        return promise.then((data) => OptInCurrentUserResponse.decode(data));
    }
    OptOutCurrentUser(request) {
        const data = Empty.encode(request).finish();
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'OptOutCurrentUser', 'application/protobuf', data);
        return promise.then((data) => OptOutCurrentUserResponse.decode(data));
    }
    ListOptedInUsers(request) {
        const data = ListOptedInUsersRequest.encode(request).finish();
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'ListOptedInUsers', 'application/protobuf', data);
        return promise.then((data) => ListOptedInUsersResponse.decode(data));
    }
    IsOptedIn(request) {
        const data = IsOptedInRequest.encode(request).finish();
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'IsOptedIn', 'application/protobuf', data);
        return promise.then((data) => IsOptedInResponse.decode(data));
    }
    Enqueue(request) {
        const data = EnqueueRequest.encode(request).finish();
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'Enqueue', 'application/protobuf', data);
        return promise.then((data) => EnqueueResponse.decode(data));
    }
    ShowGameBadge(request) {
        const data = ShowGameBadgeRequest.encode(request).finish();
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'ShowGameBadge', 'application/protobuf', data);
        return promise.then((data) => ShowGameBadgeResponse.decode(data));
    }
    DismissGameBadge(request) {
        const data = DismissGameBadgeRequest.encode(request).finish();
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'DismissGameBadge', 'application/protobuf', data);
        return promise.then((data) => DismissGameBadgeResponse.decode(data));
    }
    GetGameBadgeStatus(request) {
        const data = GetGameBadgeStatusRequest.encode(request).finish();
        const promise = this.rpc.request('devvit.plugin.notifications.Notifications', 'GetGameBadgeStatus', 'application/protobuf', data);
        return promise.then((data) => GetGameBadgeStatusResponse.decode(data));
    }
}
