// @devvit/protos/types/devvit/plugin/notifications/notifications_internal_svc.twirp.js
import { TwirpServer, TwirpError, TwirpErrorCode, TwirpContentType, chainInterceptors, } from 'twirp-ts';
import { GetGameBadgeStatusForInstallationsRequest, GetGameBadgeStatusForInstallationsResponse, } from './gamesdrawerbadge_msg.js';
export var NotificationsInternalMethod;
(function (NotificationsInternalMethod) {
    NotificationsInternalMethod["GetGameBadgeStatusForInstallations"] = "GetGameBadgeStatusForInstallations";
})(NotificationsInternalMethod || (NotificationsInternalMethod = {}));
export const NotificationsInternalMethodList = [
    NotificationsInternalMethod.GetGameBadgeStatusForInstallations,
];
export function createNotificationsInternalServer(service) {
    return new TwirpServer({
        service,
        packageName: 'devvit.plugin.notifications',
        serviceName: 'NotificationsInternal',
        methodList: NotificationsInternalMethodList,
        matchRoute: matchNotificationsInternalRoute,
    });
}
function matchNotificationsInternalRoute(method, events) {
    switch (method) {
        case 'GetGameBadgeStatusForInstallations':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'GetGameBadgeStatusForInstallations' };
                await events.onMatch(ctx);
                return handleNotificationsInternalGetGameBadgeStatusForInstallationsRequest(ctx, service, data, interceptors);
            };
        default:
            events.onNotFound();
            const msg = `no handler found`;
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleNotificationsInternalGetGameBadgeStatusForInstallationsRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleNotificationsInternalGetGameBadgeStatusForInstallationsJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleNotificationsInternalGetGameBadgeStatusForInstallationsProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
async function handleNotificationsInternalGetGameBadgeStatusForInstallationsJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = GetGameBadgeStatusForInstallationsRequest.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.GetGameBadgeStatusForInstallations(ctx, inputReq);
        });
    }
    else {
        response = await service.GetGameBadgeStatusForInstallations(ctx, request);
    }
    return JSON.stringify(GetGameBadgeStatusForInstallationsResponse.toJSON(response));
}
async function handleNotificationsInternalGetGameBadgeStatusForInstallationsProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = GetGameBadgeStatusForInstallationsRequest.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.GetGameBadgeStatusForInstallations(ctx, inputReq);
        });
    }
    else {
        response = await service.GetGameBadgeStatusForInstallations(ctx, request);
    }
    return Buffer.from(GetGameBadgeStatusForInstallationsResponse.encode(response).finish());
}
