// @devvit/protos/types/devvit/plugin/notifications/notifications_internal_svc.js
/**
 * #notifications_internal_svc.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import _m0 from "protobufjs/minimal.js";
import { GetGameBadgeStatusForInstallationsRequest, GetGameBadgeStatusForInstallationsResponse, } from "./gamesdrawerbadge_msg.js";
export const NotificationsInternalServiceName = "devvit.plugin.notifications.NotificationsInternal";
export class NotificationsInternalClientImpl {
    constructor(rpc, opts) {
        this.service = opts?.service || NotificationsInternalServiceName;
        this.rpc = rpc;
        this.GetGameBadgeStatusForInstallations = this.GetGameBadgeStatusForInstallations.bind(this);
    }
    GetGameBadgeStatusForInstallations(request, metadata) {
        const data = GetGameBadgeStatusForInstallationsRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "GetGameBadgeStatusForInstallations", data, metadata);
        return promise.then((data) => GetGameBadgeStatusForInstallationsResponse.decode(_m0.Reader.create(data)));
    }
}
export const NotificationsInternalDefinition = {
    name: "NotificationsInternal",
    fullName: "devvit.plugin.notifications.NotificationsInternal",
    methods: {
        /** Get the status of game badges for installations specified by ID */
        getGameBadgeStatusForInstallations: {
            name: "GetGameBadgeStatusForInstallations",
            requestType: GetGameBadgeStatusForInstallationsRequest,
            requestStream: false,
            responseType: GetGameBadgeStatusForInstallationsResponse,
            responseStream: false,
            options: {},
        },
    },
};
