// @devvit/protos/types/devvit/dev_portal/curated_game/curated_game.js
/**
 * #curated_game.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import _m0 from "protobufjs/minimal.js";
import { Timestamp } from "../../../google/protobuf/timestamp.js";
import { messageTypeRegistry } from "../../../typeRegistry.js";
/** The curated list type for a game entry, for example NEW or POPULAR. */
export var CuratedGameCurationType;
(function (CuratedGameCurationType) {
    /** UNSPECIFIED - Unspecified curation type. Example: omitted in a malformed request. */
    CuratedGameCurationType[CuratedGameCurationType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    /** NEW - Entry appears in the "New" curated games list. */
    CuratedGameCurationType[CuratedGameCurationType["NEW"] = 1] = "NEW";
    /** POPULAR - Entry appears in the "Popular" curated games list. */
    CuratedGameCurationType[CuratedGameCurationType["POPULAR"] = 2] = "POPULAR";
    CuratedGameCurationType[CuratedGameCurationType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(CuratedGameCurationType || (CuratedGameCurationType = {}));
export function curatedGameCurationTypeFromJSON(object) {
    switch (object) {
        case 0:
        case "UNSPECIFIED":
            return CuratedGameCurationType.UNSPECIFIED;
        case 1:
        case "NEW":
            return CuratedGameCurationType.NEW;
        case 2:
        case "POPULAR":
            return CuratedGameCurationType.POPULAR;
        case -1:
        case "UNRECOGNIZED":
        default:
            return CuratedGameCurationType.UNRECOGNIZED;
    }
}
export function curatedGameCurationTypeToJSON(object) {
    switch (object) {
        case CuratedGameCurationType.UNSPECIFIED:
            return 0;
        case CuratedGameCurationType.NEW:
            return 1;
        case CuratedGameCurationType.POPULAR:
            return 2;
        case CuratedGameCurationType.UNRECOGNIZED:
        default:
            return -1;
    }
}
function createBaseCuratedGameEntry() {
    return { appId: "", appName: "", appSlug: "", curationType: 0, lastModified: undefined, rank: 0, subredditId: "" };
}
export const CuratedGameEntry = {
    $type: "devvit.dev_portal.curated_game.CuratedGameEntry",
    encode(message, writer = _m0.Writer.create()) {
        if (message.appId !== "") {
            writer.uint32(10).string(message.appId);
        }
        if (message.appName !== "") {
            writer.uint32(18).string(message.appName);
        }
        if (message.appSlug !== "") {
            writer.uint32(26).string(message.appSlug);
        }
        if (message.curationType !== 0) {
            writer.uint32(32).int32(message.curationType);
        }
        if (message.lastModified !== undefined) {
            Timestamp.encode(toTimestamp(message.lastModified), writer.uint32(42).fork()).ldelim();
        }
        if (message.rank !== 0) {
            writer.uint32(48).int32(message.rank);
        }
        if (message.subredditId !== "") {
            writer.uint32(58).string(message.subredditId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCuratedGameEntry();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.appId = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.appName = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.appSlug = reader.string();
                    continue;
                case 4:
                    if (tag !== 32) {
                        break;
                    }
                    message.curationType = reader.int32();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.lastModified = fromTimestamp(Timestamp.decode(reader, reader.uint32()));
                    continue;
                case 6:
                    if (tag !== 48) {
                        break;
                    }
                    message.rank = reader.int32();
                    continue;
                case 7:
                    if (tag !== 58) {
                        break;
                    }
                    message.subredditId = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            appId: isSet(object.appId) ? globalThis.String(object.appId) : "",
            appName: isSet(object.appName) ? globalThis.String(object.appName) : "",
            appSlug: isSet(object.appSlug) ? globalThis.String(object.appSlug) : "",
            curationType: isSet(object.curationType) ? curatedGameCurationTypeFromJSON(object.curationType) : 0,
            lastModified: isSet(object.lastModified) ? fromJsonTimestamp(object.lastModified) : undefined,
            rank: isSet(object.rank) ? globalThis.Number(object.rank) : 0,
            subredditId: isSet(object.subredditId) ? globalThis.String(object.subredditId) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.appId !== "") {
            obj.appId = message.appId;
        }
        if (message.appName !== "") {
            obj.appName = message.appName;
        }
        if (message.appSlug !== "") {
            obj.appSlug = message.appSlug;
        }
        if (message.curationType !== 0) {
            obj.curationType = curatedGameCurationTypeToJSON(message.curationType);
        }
        if (message.lastModified !== undefined) {
            obj.lastModified = message.lastModified.toISOString();
        }
        if (message.rank !== 0) {
            obj.rank = Math.round(message.rank);
        }
        if (message.subredditId !== "") {
            obj.subredditId = message.subredditId;
        }
        return obj;
    },
    create(base) {
        return CuratedGameEntry.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCuratedGameEntry();
        message.appId = object.appId ?? "";
        message.appName = object.appName ?? "";
        message.appSlug = object.appSlug ?? "";
        message.curationType = object.curationType ?? 0;
        message.lastModified = object.lastModified ?? undefined;
        message.rank = object.rank ?? 0;
        message.subredditId = object.subredditId ?? "";
        return message;
    },
};
messageTypeRegistry.set(CuratedGameEntry.$type, CuratedGameEntry);
function createBaseListCuratedGamesRequest() {
    return { curationType: undefined };
}
export const ListCuratedGamesRequest = {
    $type: "devvit.dev_portal.curated_game.ListCuratedGamesRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.curationType !== undefined) {
            writer.uint32(8).int32(message.curationType);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseListCuratedGamesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.curationType = reader.int32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            curationType: isSet(object.curationType) ? curatedGameCurationTypeFromJSON(object.curationType) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.curationType !== undefined) {
            obj.curationType = curatedGameCurationTypeToJSON(message.curationType);
        }
        return obj;
    },
    create(base) {
        return ListCuratedGamesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseListCuratedGamesRequest();
        message.curationType = object.curationType ?? undefined;
        return message;
    },
};
messageTypeRegistry.set(ListCuratedGamesRequest.$type, ListCuratedGamesRequest);
function createBaseListCuratedGamesResponse() {
    return { entries: [] };
}
export const ListCuratedGamesResponse = {
    $type: "devvit.dev_portal.curated_game.ListCuratedGamesResponse",
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.entries) {
            CuratedGameEntry.encode(v, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseListCuratedGamesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.entries.push(CuratedGameEntry.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            entries: globalThis.Array.isArray(object?.entries)
                ? object.entries.map((e) => CuratedGameEntry.fromJSON(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.entries?.length) {
            obj.entries = message.entries.map((e) => CuratedGameEntry.toJSON(e));
        }
        return obj;
    },
    create(base) {
        return ListCuratedGamesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseListCuratedGamesResponse();
        message.entries = object.entries?.map((e) => CuratedGameEntry.fromPartial(e)) || [];
        return message;
    },
};
messageTypeRegistry.set(ListCuratedGamesResponse.$type, ListCuratedGamesResponse);
function createBaseAddCuratedGameRequest() {
    return { appSlug: "", curationType: 0, subredditId: "" };
}
export const AddCuratedGameRequest = {
    $type: "devvit.dev_portal.curated_game.AddCuratedGameRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.appSlug !== "") {
            writer.uint32(10).string(message.appSlug);
        }
        if (message.curationType !== 0) {
            writer.uint32(16).int32(message.curationType);
        }
        if (message.subredditId !== "") {
            writer.uint32(26).string(message.subredditId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAddCuratedGameRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.appSlug = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.curationType = reader.int32();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.subredditId = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            appSlug: isSet(object.appSlug) ? globalThis.String(object.appSlug) : "",
            curationType: isSet(object.curationType) ? curatedGameCurationTypeFromJSON(object.curationType) : 0,
            subredditId: isSet(object.subredditId) ? globalThis.String(object.subredditId) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.appSlug !== "") {
            obj.appSlug = message.appSlug;
        }
        if (message.curationType !== 0) {
            obj.curationType = curatedGameCurationTypeToJSON(message.curationType);
        }
        if (message.subredditId !== "") {
            obj.subredditId = message.subredditId;
        }
        return obj;
    },
    create(base) {
        return AddCuratedGameRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseAddCuratedGameRequest();
        message.appSlug = object.appSlug ?? "";
        message.curationType = object.curationType ?? 0;
        message.subredditId = object.subredditId ?? "";
        return message;
    },
};
messageTypeRegistry.set(AddCuratedGameRequest.$type, AddCuratedGameRequest);
function createBaseAddCuratedGameResponse() {
    return { entry: undefined };
}
export const AddCuratedGameResponse = {
    $type: "devvit.dev_portal.curated_game.AddCuratedGameResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.entry !== undefined) {
            CuratedGameEntry.encode(message.entry, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAddCuratedGameResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.entry = CuratedGameEntry.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { entry: isSet(object.entry) ? CuratedGameEntry.fromJSON(object.entry) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.entry !== undefined) {
            obj.entry = CuratedGameEntry.toJSON(message.entry);
        }
        return obj;
    },
    create(base) {
        return AddCuratedGameResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseAddCuratedGameResponse();
        message.entry = (object.entry !== undefined && object.entry !== null)
            ? CuratedGameEntry.fromPartial(object.entry)
            : undefined;
        return message;
    },
};
messageTypeRegistry.set(AddCuratedGameResponse.$type, AddCuratedGameResponse);
function createBaseUpdateCuratedGameRankRequest() {
    return { appSlug: "", curationType: 0, rank: 0 };
}
export const UpdateCuratedGameRankRequest = {
    $type: "devvit.dev_portal.curated_game.UpdateCuratedGameRankRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.appSlug !== "") {
            writer.uint32(10).string(message.appSlug);
        }
        if (message.curationType !== 0) {
            writer.uint32(16).int32(message.curationType);
        }
        if (message.rank !== 0) {
            writer.uint32(24).int32(message.rank);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateCuratedGameRankRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.appSlug = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.curationType = reader.int32();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.rank = reader.int32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            appSlug: isSet(object.appSlug) ? globalThis.String(object.appSlug) : "",
            curationType: isSet(object.curationType) ? curatedGameCurationTypeFromJSON(object.curationType) : 0,
            rank: isSet(object.rank) ? globalThis.Number(object.rank) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.appSlug !== "") {
            obj.appSlug = message.appSlug;
        }
        if (message.curationType !== 0) {
            obj.curationType = curatedGameCurationTypeToJSON(message.curationType);
        }
        if (message.rank !== 0) {
            obj.rank = Math.round(message.rank);
        }
        return obj;
    },
    create(base) {
        return UpdateCuratedGameRankRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdateCuratedGameRankRequest();
        message.appSlug = object.appSlug ?? "";
        message.curationType = object.curationType ?? 0;
        message.rank = object.rank ?? 0;
        return message;
    },
};
messageTypeRegistry.set(UpdateCuratedGameRankRequest.$type, UpdateCuratedGameRankRequest);
function createBaseUpdateCuratedGameRankResponse() {
    return { entry: undefined };
}
export const UpdateCuratedGameRankResponse = {
    $type: "devvit.dev_portal.curated_game.UpdateCuratedGameRankResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.entry !== undefined) {
            CuratedGameEntry.encode(message.entry, writer.uint32(10).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateCuratedGameRankResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.entry = CuratedGameEntry.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { entry: isSet(object.entry) ? CuratedGameEntry.fromJSON(object.entry) : undefined };
    },
    toJSON(message) {
        const obj = {};
        if (message.entry !== undefined) {
            obj.entry = CuratedGameEntry.toJSON(message.entry);
        }
        return obj;
    },
    create(base) {
        return UpdateCuratedGameRankResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdateCuratedGameRankResponse();
        message.entry = (object.entry !== undefined && object.entry !== null)
            ? CuratedGameEntry.fromPartial(object.entry)
            : undefined;
        return message;
    },
};
messageTypeRegistry.set(UpdateCuratedGameRankResponse.$type, UpdateCuratedGameRankResponse);
function createBaseRemoveCuratedGameRequest() {
    return { appSlug: "", curationType: 0 };
}
export const RemoveCuratedGameRequest = {
    $type: "devvit.dev_portal.curated_game.RemoveCuratedGameRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.appSlug !== "") {
            writer.uint32(10).string(message.appSlug);
        }
        if (message.curationType !== 0) {
            writer.uint32(16).int32(message.curationType);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseRemoveCuratedGameRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.appSlug = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.curationType = reader.int32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            appSlug: isSet(object.appSlug) ? globalThis.String(object.appSlug) : "",
            curationType: isSet(object.curationType) ? curatedGameCurationTypeFromJSON(object.curationType) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.appSlug !== "") {
            obj.appSlug = message.appSlug;
        }
        if (message.curationType !== 0) {
            obj.curationType = curatedGameCurationTypeToJSON(message.curationType);
        }
        return obj;
    },
    create(base) {
        return RemoveCuratedGameRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseRemoveCuratedGameRequest();
        message.appSlug = object.appSlug ?? "";
        message.curationType = object.curationType ?? 0;
        return message;
    },
};
messageTypeRegistry.set(RemoveCuratedGameRequest.$type, RemoveCuratedGameRequest);
function toTimestamp(date) {
    const seconds = Math.trunc(date.getTime() / 1000);
    const nanos = (date.getTime() % 1000) * 1000000;
    return { seconds, nanos };
}
function fromTimestamp(t) {
    let millis = (t.seconds || 0) * 1000;
    millis += (t.nanos || 0) / 1000000;
    return new globalThis.Date(millis);
}
function fromJsonTimestamp(o) {
    if (o instanceof globalThis.Date) {
        return o;
    }
    else if (typeof o === "string") {
        return new globalThis.Date(o);
    }
    else {
        return fromTimestamp(Timestamp.fromJSON(o));
    }
}
function isSet(value) {
    return value !== null && value !== undefined;
}
