// @devvit/protos/types/devvit/dev_portal/bot_registration/bot_registration.js
/**
 * #bot_registration.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import _m0 from "protobufjs/minimal.js";
import { Timestamp } from "../../../google/protobuf/timestamp.js";
import { messageTypeRegistry } from "../../../typeRegistry.js";
function createBaseAgreeToBountyRequest() {
    return { id: 0 };
}
export const AgreeToBountyRequest = {
    $type: "devvit.dev_portal.bot_registration.AgreeToBountyRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.id !== 0) {
            writer.uint32(8).int32(message.id);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAgreeToBountyRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.id = reader.int32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { id: isSet(object.id) ? globalThis.Number(object.id) : 0 };
    },
    toJSON(message) {
        const obj = {};
        if (message.id !== 0) {
            obj.id = Math.round(message.id);
        }
        return obj;
    },
    create(base) {
        return AgreeToBountyRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseAgreeToBountyRequest();
        message.id = object.id ?? 0;
        return message;
    },
};
messageTypeRegistry.set(AgreeToBountyRequest.$type, AgreeToBountyRequest);
function createBaseAgreeToBountyResponse() {
    return { id: 0, agreedToBountyAt: undefined };
}
export const AgreeToBountyResponse = {
    $type: "devvit.dev_portal.bot_registration.AgreeToBountyResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.id !== 0) {
            writer.uint32(8).int32(message.id);
        }
        if (message.agreedToBountyAt !== undefined) {
            Timestamp.encode(toTimestamp(message.agreedToBountyAt), writer.uint32(34).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAgreeToBountyResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.id = reader.int32();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.agreedToBountyAt = fromTimestamp(Timestamp.decode(reader, reader.uint32()));
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            id: isSet(object.id) ? globalThis.Number(object.id) : 0,
            agreedToBountyAt: isSet(object.agreedToBountyAt) ? fromJsonTimestamp(object.agreedToBountyAt) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.id !== 0) {
            obj.id = Math.round(message.id);
        }
        if (message.agreedToBountyAt !== undefined) {
            obj.agreedToBountyAt = message.agreedToBountyAt.toISOString();
        }
        return obj;
    },
    create(base) {
        return AgreeToBountyResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseAgreeToBountyResponse();
        message.id = object.id ?? 0;
        message.agreedToBountyAt = object.agreedToBountyAt ?? undefined;
        return message;
    },
};
messageTypeRegistry.set(AgreeToBountyResponse.$type, AgreeToBountyResponse);
function toTimestamp(date) {
    const seconds = Math.trunc(date.getTime() / 1000);
    const nanos = (date.getTime() % 1000) * 1000000;
    return { seconds, nanos };
}
function fromTimestamp(t) {
    let millis = (t.seconds || 0) * 1000;
    millis += (t.nanos || 0) / 1000000;
    return new globalThis.Date(millis);
}
function fromJsonTimestamp(o) {
    if (o instanceof globalThis.Date) {
        return o;
    }
    else if (typeof o === "string") {
        return new globalThis.Date(o);
    }
    else {
        return fromTimestamp(Timestamp.fromJSON(o));
    }
}
function isSet(value) {
    return value !== null && value !== undefined;
}
