// @devvit/protos/json/devvit/dev_portal/app/info/compute_cluster.js
/**
 * #compute_cluster.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
/**
 * ComputeCluster represents the compute infrastructure cluster where an app
 * installation runs.
 */
export var ComputeCluster;
(function (ComputeCluster) {
    /** COMPUTE_CLUSTER_UNSPECIFIED - Unspecified cluster - should not be used in practice. */
    ComputeCluster[ComputeCluster["COMPUTE_CLUSTER_UNSPECIFIED"] = 0] = "COMPUTE_CLUSTER_UNSPECIFIED";
    /** COMPUTE_GO - devvit-compute-go runtime */
    ComputeCluster[ComputeCluster["COMPUTE_GO"] = 1] = "COMPUTE_GO";
    /** ELYSIUM_GCP - devvit-orchestrator, apps on GCP */
    ComputeCluster[ComputeCluster["ELYSIUM_GCP"] = 2] = "ELYSIUM_GCP";
    /** COMPUTE_GO_STAGING - devvit-compute-go staging runtime */
    ComputeCluster[ComputeCluster["COMPUTE_GO_STAGING"] = 3] = "COMPUTE_GO_STAGING";
    ComputeCluster[ComputeCluster["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ComputeCluster || (ComputeCluster = {}));
