// @devvit/protos/types/api/proto/devvit/dev_portal/v1/app.js
/**
 * #app.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import _m0 from "protobufjs/minimal.js";
import { FullAppInfo, GetAppBySlugRequest } from "../../../../../devvit/dev_portal/app/app.js";
export const AppServiceServiceName = "devvit.dev_portal.v1.AppService";
export class AppServiceClientImpl {
    constructor(rpc, opts) {
        this.service = opts?.service || AppServiceServiceName;
        this.rpc = rpc;
        this.GetBySlug = this.GetBySlug.bind(this);
    }
    GetBySlug(request, metadata) {
        const data = GetAppBySlugRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "GetBySlug", data, metadata);
        return promise.then((data) => FullAppInfo.decode(_m0.Reader.create(data)));
    }
}
export const AppServiceDefinition = {
    name: "AppService",
    fullName: "devvit.dev_portal.v1.AppService",
    methods: {
        getBySlug: {
            name: "GetBySlug",
            requestType: GetAppBySlugRequest,
            requestStream: false,
            responseType: FullAppInfo,
            responseStream: false,
            options: {},
        },
    },
};
