// @devvit/protos/types/api/proto/devvit/dev_portal/v1/app.twirp.js
import { TwirpServer, TwirpError, TwirpErrorCode, TwirpContentType, chainInterceptors, } from 'twirp-ts';
import { GetAppBySlugRequest, FullAppInfo, } from '../../../../../devvit/dev_portal/app/app.js';
export var AppServiceMethod;
(function (AppServiceMethod) {
    AppServiceMethod["GetBySlug"] = "GetBySlug";
})(AppServiceMethod || (AppServiceMethod = {}));
export const AppServiceMethodList = [AppServiceMethod.GetBySlug];
export function createAppServiceServer(service) {
    return new TwirpServer({
        service,
        packageName: 'devvit.dev_portal.v1',
        serviceName: 'AppService',
        methodList: AppServiceMethodList,
        matchRoute: matchAppServiceRoute,
    });
}
function matchAppServiceRoute(method, events) {
    switch (method) {
        case 'GetBySlug':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'GetBySlug' };
                await events.onMatch(ctx);
                return handleAppServiceGetBySlugRequest(ctx, service, data, interceptors);
            };
        default:
            events.onNotFound();
            const msg = `no handler found`;
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleAppServiceGetBySlugRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleAppServiceGetBySlugJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleAppServiceGetBySlugProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
async function handleAppServiceGetBySlugJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = GetAppBySlugRequest.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.GetBySlug(ctx, inputReq);
        });
    }
    else {
        response = await service.GetBySlug(ctx, request);
    }
    return JSON.stringify(FullAppInfo.toJSON(response));
}
async function handleAppServiceGetBySlugProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = GetAppBySlugRequest.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.GetBySlug(ctx, inputReq);
        });
    }
    else {
        response = await service.GetBySlug(ctx, request);
    }
    return Buffer.from(FullAppInfo.encode(response).finish());
}
