// @devvit/protos/types/devvit/runtime/orchestrator.js
/**
 * #orchestrator.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import _m0 from "protobufjs/minimal.js";
import { messageTypeRegistry } from "../../typeRegistry.js";
function createBaseSetDeploymentRequest() {
    return { installationId: "", linkedBundleUrl: undefined };
}
export const SetDeploymentRequest = {
    $type: "devvit.runtime.SetDeploymentRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.installationId !== "") {
            writer.uint32(10).string(message.installationId);
        }
        if (message.linkedBundleUrl !== undefined) {
            writer.uint32(18).string(message.linkedBundleUrl);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSetDeploymentRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationId = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.linkedBundleUrl = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            installationId: isSet(object.installationId) ? globalThis.String(object.installationId) : "",
            linkedBundleUrl: isSet(object.linkedBundleUrl) ? globalThis.String(object.linkedBundleUrl) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationId !== "") {
            obj.installationId = message.installationId;
        }
        if (message.linkedBundleUrl !== undefined) {
            obj.linkedBundleUrl = message.linkedBundleUrl;
        }
        return obj;
    },
    create(base) {
        return SetDeploymentRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSetDeploymentRequest();
        message.installationId = object.installationId ?? "";
        message.linkedBundleUrl = object.linkedBundleUrl ?? undefined;
        return message;
    },
};
messageTypeRegistry.set(SetDeploymentRequest.$type, SetDeploymentRequest);
function createBaseSetDeploymentResponse() {
    return { url: "" };
}
export const SetDeploymentResponse = {
    $type: "devvit.runtime.SetDeploymentResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.url !== "") {
            writer.uint32(10).string(message.url);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseSetDeploymentResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.url = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { url: isSet(object.url) ? globalThis.String(object.url) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.url !== "") {
            obj.url = message.url;
        }
        return obj;
    },
    create(base) {
        return SetDeploymentResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseSetDeploymentResponse();
        message.url = object.url ?? "";
        return message;
    },
};
messageTypeRegistry.set(SetDeploymentResponse.$type, SetDeploymentResponse);
function createBaseDeleteDeploymentRequest() {
    return { installationId: "" };
}
export const DeleteDeploymentRequest = {
    $type: "devvit.runtime.DeleteDeploymentRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.installationId !== "") {
            writer.uint32(10).string(message.installationId);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDeleteDeploymentRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.installationId = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { installationId: isSet(object.installationId) ? globalThis.String(object.installationId) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.installationId !== "") {
            obj.installationId = message.installationId;
        }
        return obj;
    },
    create(base) {
        return DeleteDeploymentRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseDeleteDeploymentRequest();
        message.installationId = object.installationId ?? "";
        return message;
    },
};
messageTypeRegistry.set(DeleteDeploymentRequest.$type, DeleteDeploymentRequest);
function createBaseDeleteDeploymentResponse() {
    return {};
}
export const DeleteDeploymentResponse = {
    $type: "devvit.runtime.DeleteDeploymentResponse",
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDeleteDeploymentResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return DeleteDeploymentResponse.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseDeleteDeploymentResponse();
        return message;
    },
};
messageTypeRegistry.set(DeleteDeploymentResponse.$type, DeleteDeploymentResponse);
export const OrchestratorServiceName = "devvit.runtime.Orchestrator";
export class OrchestratorClientImpl {
    constructor(rpc, opts) {
        this.service = opts?.service || OrchestratorServiceName;
        this.rpc = rpc;
        this.SetDeployment = this.SetDeployment.bind(this);
        this.DeleteDeployment = this.DeleteDeployment.bind(this);
    }
    SetDeployment(request, metadata) {
        const data = SetDeploymentRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "SetDeployment", data, metadata);
        return promise.then((data) => SetDeploymentResponse.decode(_m0.Reader.create(data)));
    }
    DeleteDeployment(request, metadata) {
        const data = DeleteDeploymentRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "DeleteDeployment", data, metadata);
        return promise.then((data) => DeleteDeploymentResponse.decode(_m0.Reader.create(data)));
    }
}
export const OrchestratorDefinition = {
    name: "Orchestrator",
    fullName: "devvit.runtime.Orchestrator",
    methods: {
        setDeployment: {
            name: "SetDeployment",
            requestType: SetDeploymentRequest,
            requestStream: false,
            responseType: SetDeploymentResponse,
            responseStream: false,
            options: {},
        },
        deleteDeployment: {
            name: "DeleteDeployment",
            requestType: DeleteDeploymentRequest,
            requestStream: false,
            responseType: DeleteDeploymentResponse,
            responseStream: false,
            options: {},
        },
    },
};
function isSet(value) {
    return value !== null && value !== undefined;
}
