// @devvit/protos/types/devvit/runtime/orchestrator.twirp.js
import { TwirpServer, TwirpError, TwirpErrorCode, TwirpContentType, chainInterceptors, } from 'twirp-ts';
import { SetDeploymentRequest, SetDeploymentResponse, DeleteDeploymentRequest, DeleteDeploymentResponse, } from './orchestrator.js';
export var OrchestratorMethod;
(function (OrchestratorMethod) {
    OrchestratorMethod["SetDeployment"] = "SetDeployment";
    OrchestratorMethod["DeleteDeployment"] = "DeleteDeployment";
})(OrchestratorMethod || (OrchestratorMethod = {}));
export const OrchestratorMethodList = [
    OrchestratorMethod.SetDeployment,
    OrchestratorMethod.DeleteDeployment,
];
export function createOrchestratorServer(service) {
    return new TwirpServer({
        service,
        packageName: 'devvit.runtime',
        serviceName: 'Orchestrator',
        methodList: OrchestratorMethodList,
        matchRoute: matchOrchestratorRoute,
    });
}
function matchOrchestratorRoute(method, events) {
    switch (method) {
        case 'SetDeployment':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'SetDeployment' };
                await events.onMatch(ctx);
                return handleOrchestratorSetDeploymentRequest(ctx, service, data, interceptors);
            };
        case 'DeleteDeployment':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'DeleteDeployment' };
                await events.onMatch(ctx);
                return handleOrchestratorDeleteDeploymentRequest(ctx, service, data, interceptors);
            };
        default:
            events.onNotFound();
            const msg = `no handler found`;
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleOrchestratorSetDeploymentRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleOrchestratorSetDeploymentJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleOrchestratorSetDeploymentProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleOrchestratorDeleteDeploymentRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleOrchestratorDeleteDeploymentJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleOrchestratorDeleteDeploymentProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
async function handleOrchestratorSetDeploymentJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = SetDeploymentRequest.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.SetDeployment(ctx, inputReq);
        });
    }
    else {
        response = await service.SetDeployment(ctx, request);
    }
    return JSON.stringify(SetDeploymentResponse.toJSON(response));
}
async function handleOrchestratorDeleteDeploymentJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = DeleteDeploymentRequest.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.DeleteDeployment(ctx, inputReq);
        });
    }
    else {
        response = await service.DeleteDeployment(ctx, request);
    }
    return JSON.stringify(DeleteDeploymentResponse.toJSON(response));
}
async function handleOrchestratorSetDeploymentProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = SetDeploymentRequest.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.SetDeployment(ctx, inputReq);
        });
    }
    else {
        response = await service.SetDeployment(ctx, request);
    }
    return Buffer.from(SetDeploymentResponse.encode(response).finish());
}
async function handleOrchestratorDeleteDeploymentProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = DeleteDeploymentRequest.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.DeleteDeployment(ctx, inputReq);
        });
    }
    else {
        response = await service.DeleteDeployment(ctx, request);
    }
    return Buffer.from(DeleteDeploymentResponse.encode(response).finish());
}
