// @devvit/protos/types/devvit/dev_portal/modmail_template/modmail_template.js
/**
 * #modmail_template.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import _m0 from "protobufjs/minimal.js";
import { messageTypeRegistry } from "../../../typeRegistry.js";
/** Whether a template is used for rejection or publication modmail. */
export var ModmailTemplateActionType;
(function (ModmailTemplateActionType) {
    /** MODMAIL_TEMPLATE_ACTION_TYPE_UNSPECIFIED - Unspecified action type. */
    ModmailTemplateActionType[ModmailTemplateActionType["MODMAIL_TEMPLATE_ACTION_TYPE_UNSPECIFIED"] = 0] = "MODMAIL_TEMPLATE_ACTION_TYPE_UNSPECIFIED";
    /** MODMAIL_TEMPLATE_ACTION_TYPE_REJECT - Template is sent when rejecting an app. */
    ModmailTemplateActionType[ModmailTemplateActionType["MODMAIL_TEMPLATE_ACTION_TYPE_REJECT"] = 1] = "MODMAIL_TEMPLATE_ACTION_TYPE_REJECT";
    /** MODMAIL_TEMPLATE_ACTION_TYPE_PUBLISH - Template is sent when publishing an app. */
    ModmailTemplateActionType[ModmailTemplateActionType["MODMAIL_TEMPLATE_ACTION_TYPE_PUBLISH"] = 2] = "MODMAIL_TEMPLATE_ACTION_TYPE_PUBLISH";
    ModmailTemplateActionType[ModmailTemplateActionType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(ModmailTemplateActionType || (ModmailTemplateActionType = {}));
export function modmailTemplateActionTypeFromJSON(object) {
    switch (object) {
        case 0:
        case "MODMAIL_TEMPLATE_ACTION_TYPE_UNSPECIFIED":
            return ModmailTemplateActionType.MODMAIL_TEMPLATE_ACTION_TYPE_UNSPECIFIED;
        case 1:
        case "MODMAIL_TEMPLATE_ACTION_TYPE_REJECT":
            return ModmailTemplateActionType.MODMAIL_TEMPLATE_ACTION_TYPE_REJECT;
        case 2:
        case "MODMAIL_TEMPLATE_ACTION_TYPE_PUBLISH":
            return ModmailTemplateActionType.MODMAIL_TEMPLATE_ACTION_TYPE_PUBLISH;
        case -1:
        case "UNRECOGNIZED":
        default:
            return ModmailTemplateActionType.UNRECOGNIZED;
    }
}
export function modmailTemplateActionTypeToJSON(object) {
    switch (object) {
        case ModmailTemplateActionType.MODMAIL_TEMPLATE_ACTION_TYPE_UNSPECIFIED:
            return 0;
        case ModmailTemplateActionType.MODMAIL_TEMPLATE_ACTION_TYPE_REJECT:
            return 1;
        case ModmailTemplateActionType.MODMAIL_TEMPLATE_ACTION_TYPE_PUBLISH:
            return 2;
        case ModmailTemplateActionType.UNRECOGNIZED:
        default:
            return -1;
    }
}
function createBaseModmailTemplate() {
    return { id: "", title: "", body: "", actionType: 0, groupName: "", sortOrder: 0 };
}
export const ModmailTemplate = {
    $type: "devvit.dev_portal.modmail_template.ModmailTemplate",
    encode(message, writer = _m0.Writer.create()) {
        if (message.id !== "") {
            writer.uint32(10).string(message.id);
        }
        if (message.title !== "") {
            writer.uint32(18).string(message.title);
        }
        if (message.body !== "") {
            writer.uint32(26).string(message.body);
        }
        if (message.actionType !== 0) {
            writer.uint32(32).int32(message.actionType);
        }
        if (message.groupName !== "") {
            writer.uint32(42).string(message.groupName);
        }
        if (message.sortOrder !== 0) {
            writer.uint32(48).int32(message.sortOrder);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseModmailTemplate();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.id = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.title = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.body = reader.string();
                    continue;
                case 4:
                    if (tag !== 32) {
                        break;
                    }
                    message.actionType = reader.int32();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.groupName = reader.string();
                    continue;
                case 6:
                    if (tag !== 48) {
                        break;
                    }
                    message.sortOrder = reader.int32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            id: isSet(object.id) ? globalThis.String(object.id) : "",
            title: isSet(object.title) ? globalThis.String(object.title) : "",
            body: isSet(object.body) ? globalThis.String(object.body) : "",
            actionType: isSet(object.actionType) ? modmailTemplateActionTypeFromJSON(object.actionType) : 0,
            groupName: isSet(object.groupName) ? globalThis.String(object.groupName) : "",
            sortOrder: isSet(object.sortOrder) ? globalThis.Number(object.sortOrder) : 0,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.id !== "") {
            obj.id = message.id;
        }
        if (message.title !== "") {
            obj.title = message.title;
        }
        if (message.body !== "") {
            obj.body = message.body;
        }
        if (message.actionType !== 0) {
            obj.actionType = modmailTemplateActionTypeToJSON(message.actionType);
        }
        if (message.groupName !== "") {
            obj.groupName = message.groupName;
        }
        if (message.sortOrder !== 0) {
            obj.sortOrder = Math.round(message.sortOrder);
        }
        return obj;
    },
    create(base) {
        return ModmailTemplate.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseModmailTemplate();
        message.id = object.id ?? "";
        message.title = object.title ?? "";
        message.body = object.body ?? "";
        message.actionType = object.actionType ?? 0;
        message.groupName = object.groupName ?? "";
        message.sortOrder = object.sortOrder ?? 0;
        return message;
    },
};
messageTypeRegistry.set(ModmailTemplate.$type, ModmailTemplate);
function createBaseModmailConfig() {
    return { rejectHeader: "", rejectFooter: "", publishHeader: "", publishFooter: "" };
}
export const ModmailConfig = {
    $type: "devvit.dev_portal.modmail_template.ModmailConfig",
    encode(message, writer = _m0.Writer.create()) {
        if (message.rejectHeader !== "") {
            writer.uint32(10).string(message.rejectHeader);
        }
        if (message.rejectFooter !== "") {
            writer.uint32(18).string(message.rejectFooter);
        }
        if (message.publishHeader !== "") {
            writer.uint32(26).string(message.publishHeader);
        }
        if (message.publishFooter !== "") {
            writer.uint32(34).string(message.publishFooter);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseModmailConfig();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.rejectHeader = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.rejectFooter = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.publishHeader = reader.string();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.publishFooter = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            rejectHeader: isSet(object.rejectHeader) ? globalThis.String(object.rejectHeader) : "",
            rejectFooter: isSet(object.rejectFooter) ? globalThis.String(object.rejectFooter) : "",
            publishHeader: isSet(object.publishHeader) ? globalThis.String(object.publishHeader) : "",
            publishFooter: isSet(object.publishFooter) ? globalThis.String(object.publishFooter) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.rejectHeader !== "") {
            obj.rejectHeader = message.rejectHeader;
        }
        if (message.rejectFooter !== "") {
            obj.rejectFooter = message.rejectFooter;
        }
        if (message.publishHeader !== "") {
            obj.publishHeader = message.publishHeader;
        }
        if (message.publishFooter !== "") {
            obj.publishFooter = message.publishFooter;
        }
        return obj;
    },
    create(base) {
        return ModmailConfig.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseModmailConfig();
        message.rejectHeader = object.rejectHeader ?? "";
        message.rejectFooter = object.rejectFooter ?? "";
        message.publishHeader = object.publishHeader ?? "";
        message.publishFooter = object.publishFooter ?? "";
        return message;
    },
};
messageTypeRegistry.set(ModmailConfig.$type, ModmailConfig);
function createBaseListModmailTemplatesRequest() {
    return {};
}
export const ListModmailTemplatesRequest = {
    $type: "devvit.dev_portal.modmail_template.ListModmailTemplatesRequest",
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseListModmailTemplatesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return ListModmailTemplatesRequest.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseListModmailTemplatesRequest();
        return message;
    },
};
messageTypeRegistry.set(ListModmailTemplatesRequest.$type, ListModmailTemplatesRequest);
function createBaseListModmailTemplatesResponse() {
    return { templates: [], config: undefined };
}
export const ListModmailTemplatesResponse = {
    $type: "devvit.dev_portal.modmail_template.ListModmailTemplatesResponse",
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.templates) {
            ModmailTemplate.encode(v, writer.uint32(10).fork()).ldelim();
        }
        if (message.config !== undefined) {
            ModmailConfig.encode(message.config, writer.uint32(18).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseListModmailTemplatesResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.templates.push(ModmailTemplate.decode(reader, reader.uint32()));
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.config = ModmailConfig.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            templates: globalThis.Array.isArray(object?.templates)
                ? object.templates.map((e) => ModmailTemplate.fromJSON(e))
                : [],
            config: isSet(object.config) ? ModmailConfig.fromJSON(object.config) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.templates?.length) {
            obj.templates = message.templates.map((e) => ModmailTemplate.toJSON(e));
        }
        if (message.config !== undefined) {
            obj.config = ModmailConfig.toJSON(message.config);
        }
        return obj;
    },
    create(base) {
        return ListModmailTemplatesResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseListModmailTemplatesResponse();
        message.templates = object.templates?.map((e) => ModmailTemplate.fromPartial(e)) || [];
        message.config = (object.config !== undefined && object.config !== null)
            ? ModmailConfig.fromPartial(object.config)
            : undefined;
        return message;
    },
};
messageTypeRegistry.set(ListModmailTemplatesResponse.$type, ListModmailTemplatesResponse);
function createBaseCreateModmailTemplateRequest() {
    return { title: "", body: "", actionType: "", groupName: "" };
}
export const CreateModmailTemplateRequest = {
    $type: "devvit.dev_portal.modmail_template.CreateModmailTemplateRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.title !== "") {
            writer.uint32(10).string(message.title);
        }
        if (message.body !== "") {
            writer.uint32(18).string(message.body);
        }
        if (message.actionType !== "") {
            writer.uint32(26).string(message.actionType);
        }
        if (message.groupName !== "") {
            writer.uint32(34).string(message.groupName);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCreateModmailTemplateRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.title = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.body = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.actionType = reader.string();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.groupName = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            title: isSet(object.title) ? globalThis.String(object.title) : "",
            body: isSet(object.body) ? globalThis.String(object.body) : "",
            actionType: isSet(object.actionType) ? globalThis.String(object.actionType) : "",
            groupName: isSet(object.groupName) ? globalThis.String(object.groupName) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.title !== "") {
            obj.title = message.title;
        }
        if (message.body !== "") {
            obj.body = message.body;
        }
        if (message.actionType !== "") {
            obj.actionType = message.actionType;
        }
        if (message.groupName !== "") {
            obj.groupName = message.groupName;
        }
        return obj;
    },
    create(base) {
        return CreateModmailTemplateRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCreateModmailTemplateRequest();
        message.title = object.title ?? "";
        message.body = object.body ?? "";
        message.actionType = object.actionType ?? "";
        message.groupName = object.groupName ?? "";
        return message;
    },
};
messageTypeRegistry.set(CreateModmailTemplateRequest.$type, CreateModmailTemplateRequest);
function createBaseUpdateModmailTemplateRequest() {
    return {
        id: "",
        title: undefined,
        body: undefined,
        actionType: undefined,
        groupName: undefined,
        sortOrder: undefined,
    };
}
export const UpdateModmailTemplateRequest = {
    $type: "devvit.dev_portal.modmail_template.UpdateModmailTemplateRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.id !== "") {
            writer.uint32(10).string(message.id);
        }
        if (message.title !== undefined) {
            writer.uint32(18).string(message.title);
        }
        if (message.body !== undefined) {
            writer.uint32(26).string(message.body);
        }
        if (message.actionType !== undefined) {
            writer.uint32(34).string(message.actionType);
        }
        if (message.groupName !== undefined) {
            writer.uint32(42).string(message.groupName);
        }
        if (message.sortOrder !== undefined) {
            writer.uint32(48).int32(message.sortOrder);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateModmailTemplateRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.id = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.title = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.body = reader.string();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.actionType = reader.string();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.groupName = reader.string();
                    continue;
                case 6:
                    if (tag !== 48) {
                        break;
                    }
                    message.sortOrder = reader.int32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            id: isSet(object.id) ? globalThis.String(object.id) : "",
            title: isSet(object.title) ? globalThis.String(object.title) : undefined,
            body: isSet(object.body) ? globalThis.String(object.body) : undefined,
            actionType: isSet(object.actionType) ? globalThis.String(object.actionType) : undefined,
            groupName: isSet(object.groupName) ? globalThis.String(object.groupName) : undefined,
            sortOrder: isSet(object.sortOrder) ? globalThis.Number(object.sortOrder) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.id !== "") {
            obj.id = message.id;
        }
        if (message.title !== undefined) {
            obj.title = message.title;
        }
        if (message.body !== undefined) {
            obj.body = message.body;
        }
        if (message.actionType !== undefined) {
            obj.actionType = message.actionType;
        }
        if (message.groupName !== undefined) {
            obj.groupName = message.groupName;
        }
        if (message.sortOrder !== undefined) {
            obj.sortOrder = Math.round(message.sortOrder);
        }
        return obj;
    },
    create(base) {
        return UpdateModmailTemplateRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdateModmailTemplateRequest();
        message.id = object.id ?? "";
        message.title = object.title ?? undefined;
        message.body = object.body ?? undefined;
        message.actionType = object.actionType ?? undefined;
        message.groupName = object.groupName ?? undefined;
        message.sortOrder = object.sortOrder ?? undefined;
        return message;
    },
};
messageTypeRegistry.set(UpdateModmailTemplateRequest.$type, UpdateModmailTemplateRequest);
function createBaseDeleteModmailTemplateRequest() {
    return { id: "" };
}
export const DeleteModmailTemplateRequest = {
    $type: "devvit.dev_portal.modmail_template.DeleteModmailTemplateRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.id !== "") {
            writer.uint32(10).string(message.id);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDeleteModmailTemplateRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.id = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { id: isSet(object.id) ? globalThis.String(object.id) : "" };
    },
    toJSON(message) {
        const obj = {};
        if (message.id !== "") {
            obj.id = message.id;
        }
        return obj;
    },
    create(base) {
        return DeleteModmailTemplateRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseDeleteModmailTemplateRequest();
        message.id = object.id ?? "";
        return message;
    },
};
messageTypeRegistry.set(DeleteModmailTemplateRequest.$type, DeleteModmailTemplateRequest);
function createBaseDeleteModmailTemplateGroupRequest() {
    return { groupName: "", actionType: "", keepTemplates: false };
}
export const DeleteModmailTemplateGroupRequest = {
    $type: "devvit.dev_portal.modmail_template.DeleteModmailTemplateGroupRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.groupName !== "") {
            writer.uint32(10).string(message.groupName);
        }
        if (message.actionType !== "") {
            writer.uint32(18).string(message.actionType);
        }
        if (message.keepTemplates !== false) {
            writer.uint32(24).bool(message.keepTemplates);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseDeleteModmailTemplateGroupRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.groupName = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.actionType = reader.string();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.keepTemplates = reader.bool();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            groupName: isSet(object.groupName) ? globalThis.String(object.groupName) : "",
            actionType: isSet(object.actionType) ? globalThis.String(object.actionType) : "",
            keepTemplates: isSet(object.keepTemplates) ? globalThis.Boolean(object.keepTemplates) : false,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.groupName !== "") {
            obj.groupName = message.groupName;
        }
        if (message.actionType !== "") {
            obj.actionType = message.actionType;
        }
        if (message.keepTemplates !== false) {
            obj.keepTemplates = message.keepTemplates;
        }
        return obj;
    },
    create(base) {
        return DeleteModmailTemplateGroupRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseDeleteModmailTemplateGroupRequest();
        message.groupName = object.groupName ?? "";
        message.actionType = object.actionType ?? "";
        message.keepTemplates = object.keepTemplates ?? false;
        return message;
    },
};
messageTypeRegistry.set(DeleteModmailTemplateGroupRequest.$type, DeleteModmailTemplateGroupRequest);
function createBaseReorderModmailTemplatesRequest() {
    return { templateIds: [] };
}
export const ReorderModmailTemplatesRequest = {
    $type: "devvit.dev_portal.modmail_template.ReorderModmailTemplatesRequest",
    encode(message, writer = _m0.Writer.create()) {
        for (const v of message.templateIds) {
            writer.uint32(10).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseReorderModmailTemplatesRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.templateIds.push(reader.string());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            templateIds: globalThis.Array.isArray(object?.templateIds)
                ? object.templateIds.map((e) => globalThis.String(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.templateIds?.length) {
            obj.templateIds = message.templateIds;
        }
        return obj;
    },
    create(base) {
        return ReorderModmailTemplatesRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseReorderModmailTemplatesRequest();
        message.templateIds = object.templateIds?.map((e) => e) || [];
        return message;
    },
};
messageTypeRegistry.set(ReorderModmailTemplatesRequest.$type, ReorderModmailTemplatesRequest);
function createBaseGetModmailConfigRequest() {
    return {};
}
export const GetModmailConfigRequest = {
    $type: "devvit.dev_portal.modmail_template.GetModmailConfigRequest",
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetModmailConfigRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return GetModmailConfigRequest.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseGetModmailConfigRequest();
        return message;
    },
};
messageTypeRegistry.set(GetModmailConfigRequest.$type, GetModmailConfigRequest);
function createBaseUpdateModmailConfigRequest() {
    return { rejectHeader: undefined, rejectFooter: undefined, publishHeader: undefined, publishFooter: undefined };
}
export const UpdateModmailConfigRequest = {
    $type: "devvit.dev_portal.modmail_template.UpdateModmailConfigRequest",
    encode(message, writer = _m0.Writer.create()) {
        if (message.rejectHeader !== undefined) {
            writer.uint32(10).string(message.rejectHeader);
        }
        if (message.rejectFooter !== undefined) {
            writer.uint32(18).string(message.rejectFooter);
        }
        if (message.publishHeader !== undefined) {
            writer.uint32(26).string(message.publishHeader);
        }
        if (message.publishFooter !== undefined) {
            writer.uint32(34).string(message.publishFooter);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseUpdateModmailConfigRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.rejectHeader = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.rejectFooter = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.publishHeader = reader.string();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.publishFooter = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            rejectHeader: isSet(object.rejectHeader) ? globalThis.String(object.rejectHeader) : undefined,
            rejectFooter: isSet(object.rejectFooter) ? globalThis.String(object.rejectFooter) : undefined,
            publishHeader: isSet(object.publishHeader) ? globalThis.String(object.publishHeader) : undefined,
            publishFooter: isSet(object.publishFooter) ? globalThis.String(object.publishFooter) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.rejectHeader !== undefined) {
            obj.rejectHeader = message.rejectHeader;
        }
        if (message.rejectFooter !== undefined) {
            obj.rejectFooter = message.rejectFooter;
        }
        if (message.publishHeader !== undefined) {
            obj.publishHeader = message.publishHeader;
        }
        if (message.publishFooter !== undefined) {
            obj.publishFooter = message.publishFooter;
        }
        return obj;
    },
    create(base) {
        return UpdateModmailConfigRequest.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseUpdateModmailConfigRequest();
        message.rejectHeader = object.rejectHeader ?? undefined;
        message.rejectFooter = object.rejectFooter ?? undefined;
        message.publishHeader = object.publishHeader ?? undefined;
        message.publishFooter = object.publishFooter ?? undefined;
        return message;
    },
};
messageTypeRegistry.set(UpdateModmailConfigRequest.$type, UpdateModmailConfigRequest);
function isSet(value) {
    return value !== null && value !== undefined;
}
