// @devvit/protos/types/devvit/plugin/blob/v1alpha/blob.twirp.js
import { TwirpServer, TwirpError, TwirpErrorCode, TwirpContentType, chainInterceptors, } from 'twirp-ts';
import { GetConfigRequest, GetConfigResponse } from './blob.js';
export var BlobServiceMethod;
(function (BlobServiceMethod) {
    BlobServiceMethod["GetConfig"] = "GetConfig";
})(BlobServiceMethod || (BlobServiceMethod = {}));
export const BlobServiceMethodList = [BlobServiceMethod.GetConfig];
export function createBlobServiceServer(service) {
    return new TwirpServer({
        service,
        packageName: 'devvit.plugin.blob.v1alpha',
        serviceName: 'BlobService',
        methodList: BlobServiceMethodList,
        matchRoute: matchBlobServiceRoute,
    });
}
function matchBlobServiceRoute(method, events) {
    switch (method) {
        case 'GetConfig':
            return async (ctx, service, data, interceptors) => {
                ctx = { ...ctx, methodName: 'GetConfig' };
                await events.onMatch(ctx);
                return handleBlobServiceGetConfigRequest(ctx, service, data, interceptors);
            };
        default:
            events.onNotFound();
            const msg = `no handler found`;
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
function handleBlobServiceGetConfigRequest(ctx, service, data, interceptors) {
    switch (ctx.contentType) {
        case TwirpContentType.JSON:
            return handleBlobServiceGetConfigJSON(ctx, service, data, interceptors);
        case TwirpContentType.Protobuf:
            return handleBlobServiceGetConfigProtobuf(ctx, service, data, interceptors);
        default:
            const msg = 'unexpected Content-Type';
            throw new TwirpError(TwirpErrorCode.BadRoute, msg);
    }
}
async function handleBlobServiceGetConfigJSON(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        const body = JSON.parse(data.toString() || '{}');
        request = GetConfigRequest.fromJSON(body);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the json request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.GetConfig(ctx, inputReq);
        });
    }
    else {
        response = await service.GetConfig(ctx, request);
    }
    return JSON.stringify(GetConfigResponse.toJSON(response));
}
async function handleBlobServiceGetConfigProtobuf(ctx, service, data, interceptors) {
    let request;
    let response;
    try {
        request = GetConfigRequest.decode(data);
    }
    catch (e) {
        if (e instanceof Error) {
            const msg = 'the protobuf request could not be decoded';
            throw new TwirpError(TwirpErrorCode.Malformed, msg).withCause(e, true);
        }
    }
    if (interceptors && interceptors.length > 0) {
        const interceptor = chainInterceptors(...interceptors);
        response = await interceptor(ctx, request, (ctx, inputReq) => {
            return service.GetConfig(ctx, inputReq);
        });
    }
    else {
        response = await service.GetConfig(ctx, request);
    }
    return Buffer.from(GetConfigResponse.encode(response).finish());
}
