// @devvit/protos/types/devvit/plugin/blob/v1alpha/blob.js
/**
 * #blob.ts
 *
 * Code generated by ts-proto. DO NOT EDIT.
 * @packageDocumentation
 */
/* eslint-disable */
import _m0 from "protobufjs/minimal.js";
import { messageTypeRegistry } from "../../../../typeRegistry.js";
function createBaseGetConfigRequest() {
    return {};
}
export const GetConfigRequest = {
    $type: "devvit.plugin.blob.v1alpha.GetConfigRequest",
    encode(_, writer = _m0.Writer.create()) {
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetConfigRequest();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(_) {
        return {};
    },
    toJSON(_) {
        const obj = {};
        return obj;
    },
    create(base) {
        return GetConfigRequest.fromPartial(base ?? {});
    },
    fromPartial(_) {
        const message = createBaseGetConfigRequest();
        return message;
    },
};
messageTypeRegistry.set(GetConfigRequest.$type, GetConfigRequest);
function createBaseCredentials() {
    return { accessKey: "", secretKey: "", sessionToken: "" };
}
export const Credentials = {
    $type: "devvit.plugin.blob.v1alpha.Credentials",
    encode(message, writer = _m0.Writer.create()) {
        if (message.accessKey !== "") {
            writer.uint32(10).string(message.accessKey);
        }
        if (message.secretKey !== "") {
            writer.uint32(18).string(message.secretKey);
        }
        if (message.sessionToken !== "") {
            writer.uint32(26).string(message.sessionToken);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseCredentials();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.accessKey = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.secretKey = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.sessionToken = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            accessKey: isSet(object.accessKey) ? globalThis.String(object.accessKey) : "",
            secretKey: isSet(object.secretKey) ? globalThis.String(object.secretKey) : "",
            sessionToken: isSet(object.sessionToken) ? globalThis.String(object.sessionToken) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.accessKey !== "") {
            obj.accessKey = message.accessKey;
        }
        if (message.secretKey !== "") {
            obj.secretKey = message.secretKey;
        }
        if (message.sessionToken !== "") {
            obj.sessionToken = message.sessionToken;
        }
        return obj;
    },
    create(base) {
        return Credentials.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseCredentials();
        message.accessKey = object.accessKey ?? "";
        message.secretKey = object.secretKey ?? "";
        message.sessionToken = object.sessionToken ?? "";
        return message;
    },
};
messageTypeRegistry.set(Credentials.$type, Credentials);
function createBaseGetConfigResponse() {
    return { bucket: "", region: "", pathPrefix: "", credentials: undefined };
}
export const GetConfigResponse = {
    $type: "devvit.plugin.blob.v1alpha.GetConfigResponse",
    encode(message, writer = _m0.Writer.create()) {
        if (message.bucket !== "") {
            writer.uint32(10).string(message.bucket);
        }
        if (message.region !== "") {
            writer.uint32(18).string(message.region);
        }
        if (message.pathPrefix !== "") {
            writer.uint32(26).string(message.pathPrefix);
        }
        if (message.credentials !== undefined) {
            Credentials.encode(message.credentials, writer.uint32(34).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseGetConfigResponse();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.bucket = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.region = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.pathPrefix = reader.string();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.credentials = Credentials.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            bucket: isSet(object.bucket) ? globalThis.String(object.bucket) : "",
            region: isSet(object.region) ? globalThis.String(object.region) : "",
            pathPrefix: isSet(object.pathPrefix) ? globalThis.String(object.pathPrefix) : "",
            credentials: isSet(object.credentials) ? Credentials.fromJSON(object.credentials) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        if (message.bucket !== "") {
            obj.bucket = message.bucket;
        }
        if (message.region !== "") {
            obj.region = message.region;
        }
        if (message.pathPrefix !== "") {
            obj.pathPrefix = message.pathPrefix;
        }
        if (message.credentials !== undefined) {
            obj.credentials = Credentials.toJSON(message.credentials);
        }
        return obj;
    },
    create(base) {
        return GetConfigResponse.fromPartial(base ?? {});
    },
    fromPartial(object) {
        const message = createBaseGetConfigResponse();
        message.bucket = object.bucket ?? "";
        message.region = object.region ?? "";
        message.pathPrefix = object.pathPrefix ?? "";
        message.credentials = (object.credentials !== undefined && object.credentials !== null)
            ? Credentials.fromPartial(object.credentials)
            : undefined;
        return message;
    },
};
messageTypeRegistry.set(GetConfigResponse.$type, GetConfigResponse);
export const BlobServiceServiceName = "devvit.plugin.blob.v1alpha.BlobService";
export class BlobServiceClientImpl {
    constructor(rpc, opts) {
        this.service = opts?.service || BlobServiceServiceName;
        this.rpc = rpc;
        this.GetConfig = this.GetConfig.bind(this);
    }
    GetConfig(request, metadata) {
        const data = GetConfigRequest.encode(request).finish();
        const promise = this.rpc.request(this.service, "GetConfig", data, metadata);
        return promise.then((data) => GetConfigResponse.decode(_m0.Reader.create(data)));
    }
}
export const BlobServiceDefinition = {
    name: "BlobService",
    fullName: "devvit.plugin.blob.v1alpha.BlobService",
    methods: {
        /** Returns blob client configuration for the installation. */
        getConfig: {
            name: "GetConfig",
            requestType: GetConfigRequest,
            requestStream: false,
            responseType: GetConfigResponse,
            responseStream: false,
            options: {},
        },
    },
};
function isSet(value) {
    return value !== null && value !== undefined;
}
